--- a/linux-2.6.35.3/arch/arm/mach-mx28/mx28evk_pins.c	2012-04-18 17:54:18.674422523 -0400
+++ b/linux-2.6.35.3-dppm//arch/arm/mach-mx28/mx28evk_pins.c	2012-04-18 17:25:46.789933744 -0400
@@ -1249,6 +1249,62 @@
 	}
 }
 
+/* set all the supplied pins to input GPIOs */
+void __init mx28evk_release_pin_group(struct pin_desc *pins, unsigned count)
+{
+	int i;
+	int ret1, ret2, ret3, ret4;
+	struct pin_desc *pin;
+	for (i = 0; i < count; i++) {
+		pin = pins + i;
+		if (pin->fun == PIN_GPIO) {
+			gpio_free(MXS_PIN_TO_GPIO(pin->id));
+			ret1 = gpio_request(MXS_PIN_TO_GPIO(pin->id), pin->name);
+			ret2 = gpio_direction_input(MXS_PIN_TO_GPIO(pin->id));
+			gpio_free(MXS_PIN_TO_GPIO(pin->id));
+			printk(KERN_DEBUG "mx28evk_pin_group_release(%s): ret1: %d, ret2: %d", pin->name, ret1, ret2);
+		} else {
+			/* pin is not configured as GPIO. reconfigure as GPIO, set an input, and release */
+			mxs_release_pin(pin->id, pin->name);
+
+			ret3 = gpio_request(MXS_PIN_TO_GPIO(pin->id), pin->name);
+			ret4 = gpio_direction_input(MXS_PIN_TO_GPIO(pin->id));
+
+			gpio_free(MXS_PIN_TO_GPIO(pin->id));
+			printk(KERN_DEBUG "mx28evk_pin_group_release(%s): ret3: %d, ret4: %d\n", pin->name, ret3, ret4);
+		}
+	}
+}
+
+int
+qca_set_host_mode(uint8_t mode)
+{
+	/* disable SPI and UART */
+	mx28evk_release_pin_group(mx28evk_qca_pins_spi, ARRAY_SIZE(mx28evk_qca_pins_spi));
+	mx28evk_release_pin_group(mx28evk_qca_pins_uart, ARRAY_SIZE(mx28evk_qca_pins_uart));
+
+	switch (mode) {
+	case QCA_HOST_MODE_SPI:
+		mx28evk_init_pin_group(mx28evk_qca_pins_spi, ARRAY_SIZE(mx28evk_qca_pins_spi));
+		break;
+
+	case QCA_HOST_MODE_UART:
+		mx28evk_init_pin_group(mx28evk_qca_pins_uart, ARRAY_SIZE(mx28evk_qca_pins_uart));
+		break;
+
+	case QCA_HOST_MODE_DISABLED:
+		/* nothing to do */
+		break;
+
+	default:
+		printk("qca_set_host_mode: invalid pin mode (%d).\n", mode);
+		return -1;
+	}
+
+	return 0;
+}
+EXPORT_SYMBOL(qca_set_host_mode);
+
 #define QCA_PIN_SPI_UART_SELECT    (PINID_LCD_D12)
 #define QCA_PIN_RS232_RS485_SELECT (PINID_LCD_D11)
 #define QCA_PIN_XCVR_ENABLED       (PINID_LCD_D10)

--- a/linux-2.6.35.3/arch/arm/plat-mxs/include/mach/qca_bootstrap.h	2012-04-18 17:54:18.674422523 -0400
+++ b/linux-2.6.35.3-dppm//arch/arm/plat-mxs/include/mach/qca_bootstrap.h	2012-04-18 17:24:48.469644543 -0400
@@ -15,4 +15,6 @@
 
 uint32_t qca_read_bootstrap(void);
 
+int qca_set_host_mode(uint8_t mode);
+
 #endif
