From 394afa6a13622e4948ca4fb47cb44bfe3f653aae Mon Sep 17 00:00:00 2001
From: Nathaniel Houghton <nhoughto@qca.qualcomm.com>
Date: Wed, 16 Nov 2011 10:10:15 -0500
Subject: [PATCH 01/15] add recommended workaround for FEC lockup issues;

---
 linux-2.6.35.3/drivers/net/fec.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/linux-2.6.35.3/drivers/net/fec.c b/linux-2.6.35.3/drivers/net/fec.c
index d0e2e69..f1ddf8c 100644
--- a/linux-2.6.35.3/drivers/net/fec.c
+++ b/linux-2.6.35.3/drivers/net/fec.c
@@ -1405,9 +1405,11 @@ fec_stop(struct net_device *dev)
 			printk("fec_stop : Graceful transmit stop did not complete !\n");
 	}
 
+#if 0
 	/* Whack a reset.  We should wait for this. */
 	writel(1, fep->hwp + FEC_ECNTRL);
 	udelay(10);
+#endif
 
 #ifdef CONFIG_ARCH_MXS
 	/* Check MII or RMII */
-- 
1.7.4.1


From ffdc051f34c86212c8946377e95d6d87b80c5470 Mon Sep 17 00:00:00 2001
From: Nathaniel Houghton <nhoughto@qca.qualcomm.com>
Date: Thu, 17 Nov 2011 09:36:04 -0500
Subject: [PATCH 02/15] get the driver to a state where it can be loaded.

includes:
	- minor cleanup (use SPI_{CPOL,CPHA} instead of magic numbers)
	- reserve the I2C0_SDA pin for interrupt
	- update the spi_mxs driver to support SPI mode 3
---
 linux-2.6.35.3/arch/arm/mach-mx28/mx28evk_pins.c |   44 ++++++++++++++++++++++
 linux-2.6.35.3/drivers/spi/spi_mxs.c             |    1 +
 2 files changed, 45 insertions(+), 0 deletions(-)

diff --git a/linux-2.6.35.3/arch/arm/mach-mx28/mx28evk_pins.c b/linux-2.6.35.3/arch/arm/mach-mx28/mx28evk_pins.c
index 1990b92..0c30c38 100644
--- a/linux-2.6.35.3/arch/arm/mach-mx28/mx28evk_pins.c
+++ b/linux-2.6.35.3/arch/arm/mach-mx28/mx28evk_pins.c
@@ -164,6 +164,7 @@ static struct pin_desc mx28evk_fixed_pins[] = {
 
 #endif
 
+#if 0
 #if defined(CONFIG_I2C_MXS) || \
 	defined(CONFIG_I2C_MXS_MODULE)
 	{
@@ -183,6 +184,8 @@ static struct pin_desc mx28evk_fixed_pins[] = {
 	 .drive	= 1,
 	 },
 #endif
+#endif
+
 #if defined(CONFIG_FB_MXS) || defined(CONFIG_FB_MXS_MODULE)
 	{
 	 .name  = "LCD_D00",
@@ -1106,6 +1109,15 @@ static struct pin_desc mx28evk_spi_pins[] = {
 	 .voltage	= PAD_3_3V,
 	 .drive 	= 1,
 	 },
+	{
+	 .name = "ATHSPI INT",
+	 .id = PINID_I2C0_SDA,
+	 .fun = PIN_GPIO,
+	 .strength = PAD_8MA,
+	 .voltage = PAD_3_3V,
+	 .drive	= 1,
+	 .output = 0,
+	 },
 };
 #endif
 
@@ -1229,3 +1241,35 @@ void __init mx28evk_pins_init(void)
 						ARRAY_SIZE(mx28evk_eth_pins));
 #endif
 }
+
+/*
+ * Qualcomm Atheros SPI interrupt configuration functions.
+ */
+
+int gpio_spi_intr_cfg(void)
+{
+
+#if 0
+	int status;
+
+	status = gpio_request(MXS_PIN_TO_GPIO(PINID_I2C0_SDA), "ATHEROS");
+	if (status)
+		return status;
+
+	status = gpio_direction_input(MXS_PIN_TO_GPIO(PINID_I2C0_SDA));
+	if (status)
+		return status;
+#endif
+
+	set_irq_type(gpio_to_irq(MXS_PIN_TO_GPIO(PINID_I2C0_SDA)), IRQ_TYPE_EDGE_RISING);
+
+	return 0;
+}
+
+int gpio_spi_intr_get_irq(void)
+{
+	return gpio_to_irq(MXS_PIN_TO_GPIO(PINID_I2C0_SDA));
+}
+
+EXPORT_SYMBOL(gpio_spi_intr_cfg);
+EXPORT_SYMBOL(gpio_spi_intr_get_irq);
diff --git a/linux-2.6.35.3/drivers/spi/spi_mxs.c b/linux-2.6.35.3/drivers/spi/spi_mxs.c
index c6e66fd..16c353c 100644
--- a/linux-2.6.35.3/drivers/spi/spi_mxs.c
+++ b/linux-2.6.35.3/drivers/spi/spi_mxs.c
@@ -559,6 +559,7 @@ static int __init mxs_spi_probe(struct platform_device *dev)
 	master->transfer = mxs_spi_transfer;
 	master->setup = mxs_spi_setup;
 	master->cleanup = mxs_spi_cleanup;
+	master->mode_bits = MODEBITS;
 
 	if (!request_mem_region(r->start,
 				resource_size(r), dev_name(&dev->dev))) {
-- 
1.7.4.1


From 4be64aa23cdd81c99152edc14b33d2ded982b342 Mon Sep 17 00:00:00 2001
From: Nathaniel Houghton <nhoughto@qca.qualcomm.com>
Date: Wed, 23 Nov 2011 13:40:02 -0500
Subject: [PATCH 03/15] re-enable promiscuous mode after link restart;

---
 linux-2.6.35.3/drivers/net/fec.c |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/linux-2.6.35.3/drivers/net/fec.c b/linux-2.6.35.3/drivers/net/fec.c
index f1ddf8c..93de97a 100644
--- a/linux-2.6.35.3/drivers/net/fec.c
+++ b/linux-2.6.35.3/drivers/net/fec.c
@@ -1383,6 +1383,14 @@ fec_restart(struct net_device *dev, int duplex)
 	}
 #endif
 
+	/* Re-enable promiscuous mode if needed. */
+	if (dev->flags & IFF_PROMISC) {
+		unsigned int tmp;
+		tmp = readl(fep->hwp + FEC_R_CNTRL);
+		tmp |= 0x8;
+		writel(tmp, fep->hwp + FEC_R_CNTRL);
+	}
+
 	/* And last, enable the transmit and receive processing */
 	reg |= 0x00000002;
 	writel(reg, fep->hwp + FEC_ECNTRL);
-- 
1.7.4.1


From 9375156f8aeb852b158395e5a16e4335aa5e5491 Mon Sep 17 00:00:00 2001
From: Nathaniel Houghton <nhoughto@qca.qualcomm.com>
Date: Wed, 23 Nov 2011 13:58:22 -0500
Subject: [PATCH 04/15] remove #if 0 block.

---
 linux-2.6.35.3/arch/arm/mach-mx28/mx28evk_pins.c |   13 -------------
 1 files changed, 0 insertions(+), 13 deletions(-)

diff --git a/linux-2.6.35.3/arch/arm/mach-mx28/mx28evk_pins.c b/linux-2.6.35.3/arch/arm/mach-mx28/mx28evk_pins.c
index 0c30c38..6f00afd 100644
--- a/linux-2.6.35.3/arch/arm/mach-mx28/mx28evk_pins.c
+++ b/linux-2.6.35.3/arch/arm/mach-mx28/mx28evk_pins.c
@@ -1248,19 +1248,6 @@ void __init mx28evk_pins_init(void)
 
 int gpio_spi_intr_cfg(void)
 {
-
-#if 0
-	int status;
-
-	status = gpio_request(MXS_PIN_TO_GPIO(PINID_I2C0_SDA), "ATHEROS");
-	if (status)
-		return status;
-
-	status = gpio_direction_input(MXS_PIN_TO_GPIO(PINID_I2C0_SDA));
-	if (status)
-		return status;
-#endif
-
 	set_irq_type(gpio_to_irq(MXS_PIN_TO_GPIO(PINID_I2C0_SDA)), IRQ_TYPE_EDGE_RISING);
 
 	return 0;
-- 
1.7.4.1


From ad71754c427c202d15758296fbe6d44d023cab3a Mon Sep 17 00:00:00 2001
From: Nathaniel Houghton <nhoughto@qca.qualcomm.com>
Date: Mon, 28 Nov 2011 18:05:40 -0500
Subject: [PATCH 05/15] update .gitignore;

---
 linux-2.6.35.3/arch/arm/boot/compressed/.gitignore |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/linux-2.6.35.3/arch/arm/boot/compressed/.gitignore b/linux-2.6.35.3/arch/arm/boot/compressed/.gitignore
index ab204db..c52a6dd 100644
--- a/linux-2.6.35.3/arch/arm/boot/compressed/.gitignore
+++ b/linux-2.6.35.3/arch/arm/boot/compressed/.gitignore
@@ -1,3 +1,6 @@
 font.c
 piggy.gz
 vmlinux.lds
+lib1funcs.S
+piggy.gzip
+vmlinux
-- 
1.7.4.1


From bf2d097cd3d28fcf4eaf34d39b5291575eb8dad9 Mon Sep 17 00:00:00 2001
From: Nathaniel Houghton <nhoughto@qca.qualcomm.com>
Date: Fri, 2 Dec 2011 15:54:44 -0500
Subject: [PATCH 06/15] update link up/down fix for FEC;

---
 linux-2.6.35.3/drivers/net/fec.c |    8 +++++---
 1 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/linux-2.6.35.3/drivers/net/fec.c b/linux-2.6.35.3/drivers/net/fec.c
index 93de97a..7b2ad95 100644
--- a/linux-2.6.35.3/drivers/net/fec.c
+++ b/linux-2.6.35.3/drivers/net/fec.c
@@ -121,8 +121,10 @@
 #if defined(CONFIG_FEC_1588) && defined(CONFIG_ARCH_MX28)
 #define FEC_DEFAULT_IMASK (FEC_ENET_TXF | FEC_ENET_RXF | FEC_ENET_MII | \
 				FEC_ENET_TS_AVAIL | FEC_ENET_TS_TIMER)
+#define FEC_STOP_IMASK (FEC_ENET_MII)
 #else
 #define FEC_DEFAULT_IMASK (FEC_ENET_TXF | FEC_ENET_RXF | FEC_ENET_MII)
+#define FEC_STOP_IMASK (FEC_ENET_MII)
 #endif
 
 /* The FEC stores dest/src/type, data, and checksum for receive packets.
@@ -1413,11 +1415,11 @@ fec_stop(struct net_device *dev)
 			printk("fec_stop : Graceful transmit stop did not complete !\n");
 	}
 
-#if 0
 	/* Whack a reset.  We should wait for this. */
 	writel(1, fep->hwp + FEC_ECNTRL);
 	udelay(10);
-#endif
+
+	writel(0x00000002, fep->hwp + FEC_ECNTRL);
 
 #ifdef CONFIG_ARCH_MXS
 	/* Check MII or RMII */
@@ -1433,7 +1435,7 @@ fec_stop(struct net_device *dev)
 	writel(fep->phy_speed, fep->hwp + FEC_MII_SPEED);
 	if (fep->ptimer_present)
 		fec_ptp_stop(fep->ptp_priv);
-	writel(FEC_DEFAULT_IMASK, fep->hwp + FEC_IMASK);
+	writel(FEC_STOP_IMASK, fep->hwp + FEC_IMASK);
 
 	netif_stop_queue(dev);
 	fep->link = 0;
-- 
1.7.4.1


From 261a1d89227c42e5dafee6390b41fc2f86c568b2 Mon Sep 17 00:00:00 2001
From: Nathaniel Houghton <nhoughto@qca.qualcomm.com>
Date: Fri, 2 Dec 2011 15:55:22 -0500
Subject: [PATCH 07/15] update FEC driver to copy the SKB if it needs to swap the data bytes;

---
 linux-2.6.35.3/drivers/net/fec.c |   14 ++++++++++++++
 1 files changed, 14 insertions(+), 0 deletions(-)

diff --git a/linux-2.6.35.3/drivers/net/fec.c b/linux-2.6.35.3/drivers/net/fec.c
index 7b2ad95..b980d70 100644
--- a/linux-2.6.35.3/drivers/net/fec.c
+++ b/linux-2.6.35.3/drivers/net/fec.c
@@ -277,6 +277,20 @@ fec_enet_start_xmit(struct sk_buff *skb, struct net_device *dev)
 	/* Clear all of the status flags */
 	status &= ~BD_ENET_TX_STATS;
 
+#ifdef CONFIG_ARCH_MXS
+{
+	/* make a copy of the skb if we need to swap the bytes */
+	struct sk_buff *tskb;
+	tskb = skb_copy(skb, GFP_ATOMIC);
+	if (tskb == NULL) {
+		spin_unlock_irqrestore(&fep->hw_lock, flags);
+		return NETDEV_TX_BUSY;
+	}
+	dev_kfree_skb(skb);
+	skb = tskb;
+}
+#endif
+
 	/* Set buffer length and buffer pointer */
 	bufaddr = skb->data;
 	bdp->cbd_datlen = skb->len;
-- 
1.7.4.1


From 2a187471d48cf100cce38802173cc2215215c217 Mon Sep 17 00:00:00 2001
From: Nathaniel Houghton <nhoughto@qca.qualcomm.com>
Date: Fri, 2 Dec 2011 18:39:15 -0500
Subject: [PATCH 08/15] use PIO for 4 byte or less transfers;

---
 linux-2.6.35.3/drivers/spi/spi_mxs.c |   36 +++++++++++++++++++++++----------
 1 files changed, 25 insertions(+), 11 deletions(-)

diff --git a/linux-2.6.35.3/drivers/spi/spi_mxs.c b/linux-2.6.35.3/drivers/spi/spi_mxs.c
index 16c353c..1c36db9 100644
--- a/linux-2.6.35.3/drivers/spi/spi_mxs.c
+++ b/linux-2.6.35.3/drivers/spi/spi_mxs.c
@@ -157,13 +157,23 @@ static int mxs_spi_setup_transfer(struct spi_device *spi,
 		     BF_SSP_TIMING_CLOCK_RATE(rate - 1),
 		     ss->regs + HW_SSP_TIMING);
 
-	__raw_writel(BF_SSP_CTRL1_SSP_MODE(BV_SSP_CTRL1_SSP_MODE__SPI) |
-		     BF_SSP_CTRL1_WORD_LENGTH
-		     (BV_SSP_CTRL1_WORD_LENGTH__EIGHT_BITS) |
-		     ((spi->mode & SPI_CPOL) ? BM_SSP_CTRL1_POLARITY : 0) |
-		     ((spi->mode & SPI_CPHA) ? BM_SSP_CTRL1_PHASE : 0) |
-		     (pio ? 0 : BM_SSP_CTRL1_DMA_ENABLE),
-		     ss->regs + HW_SSP_CTRL1);
+	if (pio || (t && t->len <= 4)) {
+		__raw_writel(BF_SSP_CTRL1_SSP_MODE(BV_SSP_CTRL1_SSP_MODE__SPI) |
+			     BF_SSP_CTRL1_WORD_LENGTH
+			     (BV_SSP_CTRL1_WORD_LENGTH__EIGHT_BITS) |
+			     ((spi->mode & SPI_CPOL) ? BM_SSP_CTRL1_POLARITY : 0) |
+			     ((spi->mode & SPI_CPHA) ? BM_SSP_CTRL1_PHASE : 0) |
+			     0,
+			     ss->regs + HW_SSP_CTRL1);
+	} else {
+		__raw_writel(BF_SSP_CTRL1_SSP_MODE(BV_SSP_CTRL1_SSP_MODE__SPI) |
+			     BF_SSP_CTRL1_WORD_LENGTH
+			     (BV_SSP_CTRL1_WORD_LENGTH__EIGHT_BITS) |
+			     ((spi->mode & SPI_CPOL) ? BM_SSP_CTRL1_POLARITY : 0) |
+			     ((spi->mode & SPI_CPHA) ? BM_SSP_CTRL1_PHASE : 0) |
+			     BM_SSP_CTRL1_DMA_ENABLE,
+			     ss->regs + HW_SSP_CTRL1);
+	}
 
 	__raw_writel(0x00, ss->regs + HW_SSP_CMD0_SET);
 
@@ -397,12 +407,14 @@ static int mxs_spi_handle_message(struct mxs_spi *ss, struct spi_message *m)
 		   here driver completely ignores setting of t->cs_change
 		 */
 		if (t->tx_buf) {
-			status = pio ?
+			if (pio || t->len <= 4) {
 			    mxs_spi_txrx_pio(ss, cs, (void *)t->tx_buf,
-					     t->len, &first, &last, 1) :
+					     t->len, &first, &last, 1);
+			} else {
 			    mxs_spi_txrx_dma(ss, cs, (void *)t->tx_buf,
 					     t->tx_dma, t->len, &first, &last,
 					     1);
+			}
 			if (debug) {
 				if (t->len < 0x10)
 					print_hex_dump_bytes("Tx ",
@@ -413,12 +425,14 @@ static int mxs_spi_handle_message(struct mxs_spi *ss, struct spi_message *m)
 			}
 		}
 		if (t->rx_buf) {
-			status = pio ?
+			if (pio || t->len <= 4) {
 			    mxs_spi_txrx_pio(ss, cs, t->rx_buf,
-					     t->len, &first, &last, 0) :
+					     t->len, &first, &last, 0);
+			} else {
 			    mxs_spi_txrx_dma(ss, cs, t->rx_buf,
 					     t->rx_dma, t->len, &first, &last,
 					     0);
+			}
 			if (debug) {
 				if (t->len < 0x10)
 					print_hex_dump_bytes("Rx ",
-- 
1.7.4.1


From 7d2cf6d0f755af4f223ca4531c68b2a43cdcd615 Mon Sep 17 00:00:00 2001
From: Nathaniel Houghton <nhoughto@qca.qualcomm.com>
Date: Wed, 7 Dec 2011 13:33:45 -0500
Subject: [PATCH 09/15] update PIN configuration for PL16 board;

---
 linux-2.6.35.3/arch/arm/mach-mx28/mx28evk_pins.c |  320 +++++++++++++---------
 1 files changed, 186 insertions(+), 134 deletions(-)

diff --git a/linux-2.6.35.3/arch/arm/mach-mx28/mx28evk_pins.c b/linux-2.6.35.3/arch/arm/mach-mx28/mx28evk_pins.c
index 6f00afd..7079be4 100644
--- a/linux-2.6.35.3/arch/arm/mach-mx28/mx28evk_pins.c
+++ b/linux-2.6.35.3/arch/arm/mach-mx28/mx28evk_pins.c
@@ -38,28 +38,6 @@ static struct pin_desc mx28evk_fixed_pins[] = {
 	 .id = PINID_PWM1,
 	 .fun = PIN_FUN3,
 	 },
-#ifdef CONFIG_MXS_AUART0_DEVICE_ENABLE
-	{
-	 .name  = "AUART0.RX",
-	 .id    = PINID_AUART0_RX,
-	 .fun   = PIN_FUN1,
-	 },
-	{
-	 .name  = "AUART0.TX",
-	 .id    = PINID_AUART0_TX,
-	 .fun   = PIN_FUN1,
-	 },
-	{
-	 .name  = "AUART0.CTS",
-	 .id    = PINID_AUART0_CTS,
-	 .fun   = PIN_FUN1,
-	 },
-	{
-	 .name  = "AUART0.RTS",
-	 .id    = PINID_AUART0_RTS,
-	 .fun   = PIN_FUN1,
-	 },
-#endif
 #ifdef CONFIG_MXS_AUART3_DEVICE_ENABLE
 	{
 	 .name  = "AUART3.RX",
@@ -164,7 +142,6 @@ static struct pin_desc mx28evk_fixed_pins[] = {
 
 #endif
 
-#if 0
 #if defined(CONFIG_I2C_MXS) || \
 	defined(CONFIG_I2C_MXS_MODULE)
 	{
@@ -175,15 +152,6 @@ static struct pin_desc mx28evk_fixed_pins[] = {
 	 .voltage = PAD_3_3V,
 	 .drive	= 1,
 	 },
-	{
-	 .name = "I2C0_SDA",
-	 .id = PINID_I2C0_SDA,
-	 .fun = PIN_FUN1,
-	 .strength = PAD_8MA,
-	 .voltage = PAD_3_3V,
-	 .drive	= 1,
-	 },
-#endif
 #endif
 
 #if defined(CONFIG_FB_MXS) || defined(CONFIG_FB_MXS_MODULE)
@@ -244,14 +212,6 @@ static struct pin_desc mx28evk_fixed_pins[] = {
 	 .drive	= 1,
 	 },
 	{
-	 .name  = "LCD_D07",
-	 .id	= PINID_LCD_D07,
-	 .fun	= PIN_FUN1,
-	 .strength = PAD_8MA,
-	 .voltage	= PAD_3_3V,
-	 .drive	= 1,
-	 },
-	{
 	 .name  = "LCD_D08",
 	 .id	= PINID_LCD_D08,
 	 .fun	= PIN_FUN1,
@@ -268,38 +228,6 @@ static struct pin_desc mx28evk_fixed_pins[] = {
 	 .drive	= 1,
 	 },
 	{
-	 .name  = "LCD_D10",
-	 .id	= PINID_LCD_D10,
-	 .fun	= PIN_FUN1,
-	 .strength = PAD_8MA,
-	 .voltage	= PAD_3_3V,
-	 .drive	= 1,
-	 },
-	{
-	 .name  = "LCD_D11",
-	 .id	= PINID_LCD_D11,
-	 .fun	= PIN_FUN1,
-	 .strength = PAD_8MA,
-	 .voltage	= PAD_3_3V,
-	 .drive	= 1,
-	 },
-	{
-	 .name  = "LCD_D12",
-	 .id	= PINID_LCD_D12,
-	 .fun	= PIN_FUN1,
-	 .strength = PAD_8MA,
-	 .voltage	= PAD_3_3V,
-	 .drive	= 1,
-	 },
-	{
-	 .name  = "LCD_D13",
-	 .id	= PINID_LCD_D13,
-	 .fun	= PIN_FUN1,
-	 .strength = PAD_8MA,
-	 .voltage	= PAD_3_3V,
-	 .drive	= 1,
-	 },
-	{
 	 .name  = "LCD_D14",
 	 .id	= PINID_LCD_D14,
 	 .fun	= PIN_FUN1,
@@ -388,22 +316,6 @@ static struct pin_desc mx28evk_fixed_pins[] = {
 	 .drive	= 1,
 	 },
 	{
-	 .name = "LCD_VSYNC",
-	 .id   = PINID_LCD_RD_E,
-	 .fun  = PIN_FUN2,
-	 .strength = PAD_8MA,
-	 .voltage = PAD_3_3V,
-	 .drive	= 1,
-	 },
-	{
-	 .name = "LCD_HSYNC",
-	 .id = PINID_LCD_WR_RWN,
-	 .fun = PIN_FUN2,
-	 .strength = PAD_8MA,
-	 .voltage = PAD_3_3V,
-	 .drive	= 1,
-	 },
-	{
 	 .name = "LCD_ENABLE",
 	 .id = PINID_LCD_CS,
 	 .fun = PIN_FUN2,
@@ -412,14 +324,6 @@ static struct pin_desc mx28evk_fixed_pins[] = {
 	 .drive	= 1,
 	 },
 	{
-	 .name = "LCD_DOTCLK",
-	 .id = PINID_LCD_RS,
-	 .fun = PIN_FUN2,
-	 .strength = PAD_8MA,
-	 .voltage = PAD_3_3V,
-	 .drive	= 1,
-	 },
-	{
 	 .name = "LCD_BACKLIGHT",
 	 .id = PINID_PWM2,
 	 .fun = PIN_FUN1,
@@ -521,16 +425,6 @@ static struct pin_desc mx28evk_fixed_pins[] = {
 	 .pull 		= 1,
 	 },
 	{
-	 .name	= "SSP0_DETECT",
-	 .id	= PINID_SSP0_DETECT,
-	 .fun	= PIN_FUN1,
-	 .strength	= PAD_8MA,
-	 .voltage	= PAD_3_3V,
-	 .pullup	= 0,
-	 .drive 	= 1,
-	 .pull 		= 0,
-	 },
-	{
 	 .name	= "SSP0_SCK",
 	 .id	= PINID_SSP0_SCK,
 	 .fun	= PIN_FUN1,
@@ -888,16 +782,6 @@ static struct pin_desc mx28evk_ssp1_pins[] = {
 	 .pull 		= 1,
 	 },
 	{
-	 .name	= "SSP1_DETECT",
-	 .id	= PINID_GPMI_RDY0,
-	 .fun	= PIN_FUN1,
-	 .strength	= PAD_8MA,
-	 .voltage	= PAD_3_3V,
-	 .pullup	= 0,
-	 .drive 	= 1,
-	 .pull 		= 0,
-	 },
-	{
 	 .name	= "SSP1_SCK",
 	 .id	= PINID_GPMI_WRN,
 	 .fun	= PIN_FUN2,
@@ -1011,15 +895,6 @@ static struct pin_desc mx28evk_gpmi_pins[] = {
 	 .drive    = !0
 	 },
 	{
-	 .name     = "GPMI RDY0",
-	 .id       = PINID_GPMI_RDY0,
-	 .fun      = PIN_FUN1,
-	 .strength = PAD_4MA,
-	 .voltage  = PAD_3_3V,
-	 .pullup   = 0,
-	 .drive    = !0
-	 },
-	{
 	 .name     = "GPMI RDY1",
 	 .id       = PINID_GPMI_RDY1,
 	 .fun      = PIN_FUN1,
@@ -1075,8 +950,8 @@ static struct pin_desc mx28evk_gpmi_pins[] = {
 	 },
 };
 
-#if defined(CONFIG_SPI_MXS) || defined(CONFIG_SPI_MXS_MODULE)
-static struct pin_desc mx28evk_spi_pins[] = {
+static struct pin_desc mx28evk_atheros_pins[] = {
+	/* SPI interface */
 	{
 	 .name	= "SSP2 MOSI",
 	 .id	= PINID_SSP2_MOSI,
@@ -1109,18 +984,197 @@ static struct pin_desc mx28evk_spi_pins[] = {
 	 .voltage	= PAD_3_3V,
 	 .drive 	= 1,
 	 },
+
+	/* QCA7000 interrupt lines */
 	{
-	 .name = "ATHSPI INT",
+	 .name = "QCA7K_INT0",
 	 .id = PINID_I2C0_SDA,
 	 .fun = PIN_GPIO,
-	 .strength = PAD_8MA,
+	 .strength = PAD_4MA,
 	 .voltage = PAD_3_3V,
 	 .drive	= 1,
 	 .output = 0,
 	 },
-};
+	{
+	 .name = "QCA7K_INT1",
+	 .id	= PINID_GPMI_RDY0,
+	 .fun	= PIN_GPIO,
+	 .strength = PAD_4MA,
+	 .voltage = PAD_3_3V,
+	 .drive	= 1,
+	 .output = 0,
+	 },
+	{
+	 .name = "QCA7K_INT2",
+	 .id = PINID_SSP0_DETECT,
+	 .fun = PIN_GPIO,
+	 .strength = PAD_4MA,
+	 .voltage = PAD_3_3V,
+	 .drive	= 1,
+	 .output = 0,
+	 },
+
+	/* QCA7000 Reset */
+	{
+	 .name  = "QCA7K_RESET",
+	 .id	= PINID_LCD_D13,
+	 .fun	= PIN_GPIO,
+	 .strength = PAD_4MA,
+	 .voltage = PAD_3_3V,
+	 .drive	= 1,
+	 .output = 0,
+	 },
+
+	/* QCA7000 GPIOs */
+	{
+	 .name = "QCA7K_GPIO_0",
+	 .id = PINID_LCD_WR_RWN,
+	 .fun = PIN_GPIO,
+	 .strength = PAD_4MA,
+	 .voltage = PAD_3_3V,
+	 .drive	= 1,
+	 .output = 0,
+	 },
+	{
+	 .name = "QCA7K_GPIO_1",
+	 .id   = PINID_LCD_VSYNC,
+	 .fun  = PIN_GPIO,
+	 .strength = PAD_4MA,
+	 .voltage = PAD_3_3V,
+	 .drive	= 1,
+	 .output = 0,
+	 },
+	{
+	 .name = "QCA7K_GPIO_2",
+	 .id = PINID_LCD_HSYNC,
+	 .fun = PIN_GPIO,
+	 .strength = PAD_4MA,
+	 .voltage = PAD_3_3V,
+	 .drive	= 1,
+	 .output = 0,
+	 },
+	{
+	 .name = "QCA7K_GPIO_3",
+	 .id = PINID_LCD_DOTCK,
+	 .fun = PIN_GPIO,
+	 .strength = PAD_4MA,
+	 .voltage = PAD_3_3V,
+	 .drive	= 1,
+	 .output = 0,
+	 },
+
+	 /* QCA7000 UART interface */
+#if 0
+	{
+	 .name  = "AUART0.RX",
+	 .id    = PINID_AUART0_RX,
+	 .fun   = PIN_FUN1,
+	 .strength = PAD_4MA,
+	 .voltage = PAD_3_3V,
+	 .drive = 1,
+	 },
+	{
+	 .name  = "AUART0.TX",
+	 .id    = PINID_AUART0_TX,
+	 .fun   = PIN_FUN1,
+	 .strength = PAD_4MA,
+	 .voltage = PAD_3_3V,
+	 .drive = 1,
+	 },
+	{
+	 .name  = "AUART0.CTS",
+	 .id    = PINID_AUART0_CTS,
+	 .fun   = PIN_FUN1,
+	 .strength = PAD_4MA,
+	 .voltage = PAD_3_3V,
+	 .drive = 1,
+	 },
+	{
+	 .name  = "AUART0.RTS",
+	 .id    = PINID_AUART0_RTS,
+	 .fun   = PIN_FUN1,
+	 .strength = PAD_4MA,
+	 .voltage = PAD_3_3V,
+	 .drive = 1,
+	 },
+#else
+	{
+	 .name  = "AUART0.RX",
+	 .id    = PINID_AUART0_RX,
+	 .fun   = PIN_GPIO,
+	 .strength = PAD_4MA,
+	 .voltage = PAD_3_3V,
+	 .drive = 1,
+	 .output = 0
+	 },
+	{
+	 .name  = "AUART0.TX",
+	 .id    = PINID_AUART0_TX,
+	 .fun   = PIN_GPIO,
+	 .strength = PAD_4MA,
+	 .voltage = PAD_3_3V,
+	 .drive = 1,
+	 .output = 0
+	 },
+	{
+	 .name  = "AUART0.CTS",
+	 .id    = PINID_AUART0_CTS,
+	 .fun   = PIN_GPIO,
+	 .strength = PAD_4MA,
+	 .voltage = PAD_3_3V,
+	 .drive = 1,
+	 .output = 0
+	 },
+	{
+	 .name  = "AUART0.RTS",
+	 .id    = PINID_AUART0_RTS,
+	 .fun   = PIN_GPIO,
+	 .strength = PAD_4MA,
+	 .voltage = PAD_3_3V,
+	 .drive = 1,
+	 .output = 0
+	 },
 #endif
 
+	/* Configuration SW13 */
+	{
+	 .name  = "SW13 0",
+	 .id	= PINID_LCD_D07,
+	 .fun	= PIN_GPIO,
+	 .strength = PAD_4MA,
+	 .voltage = PAD_3_3V,
+	 .drive	= 1,
+	 .output = 0,
+	 },
+	{
+	 .name  = "SW13 1",
+	 .id	= PINID_LCD_D10,
+	 .fun	= PIN_GPIO,
+	 .strength = PAD_4MA,
+	 .voltage = PAD_3_3V,
+	 .drive	= 1,
+	 .output = 0,
+	 },
+	{
+	 .name  = "SW13 2",
+	 .id	= PINID_LCD_D11,
+	 .fun	= PIN_GPIO,
+	 .strength = PAD_4MA,
+	 .voltage = PAD_3_3V,
+	 .drive	= 1,
+	 .output = 0,
+	 },
+	{
+	 .name  = "SW13 3",
+	 .id	= PINID_LCD_D12,
+	 .fun	= PIN_GPIO,
+	 .strength = PAD_4MA,
+	 .voltage = PAD_3_3V,
+	 .drive	= 1,
+	 .output = 0,
+	 },
+};
+
 #if defined(CONFIG_FEC) || defined(CONFIG_FEC_MODULE)\
 	|| defined(CONFIG_FEC_L2SWITCH)
 int mx28evk_enet_gpio_init(void)
@@ -1230,10 +1284,8 @@ void __init mx28evk_pins_init(void)
 						ARRAY_SIZE(mx28evk_gpmi_pins));
 	}
 
-#if defined(CONFIG_SPI_MXS) || defined(CONFIG_SPI_MXS_MODULE)
-	mx28evk_init_pin_group(mx28evk_spi_pins,
-					ARRAY_SIZE(mx28evk_spi_pins));
-#endif
+	mx28evk_init_pin_group(mx28evk_atheros_pins,
+					ARRAY_SIZE(mx28evk_atheros_pins));
 
 #if defined(CONFIG_FEC) || defined(CONFIG_FEC_MODULE)\
 	|| defined(CONFIG_FEC_L2SWITCH)
-- 
1.7.4.1


From dcf6c3b46b945c02b529a079277070d872419bd7 Mon Sep 17 00:00:00 2001
From: Nathaniel Houghton <nhoughto@qca.qualcomm.com>
Date: Mon, 12 Dec 2011 10:08:36 -0500
Subject: [PATCH 10/15] update pin configuration;

---
 linux-2.6.35.3/arch/arm/mach-mx28/mx28evk_pins.c |  213 ++++++++++++----------
 1 files changed, 119 insertions(+), 94 deletions(-)

diff --git a/linux-2.6.35.3/arch/arm/mach-mx28/mx28evk_pins.c b/linux-2.6.35.3/arch/arm/mach-mx28/mx28evk_pins.c
index 7079be4..1476768 100644
--- a/linux-2.6.35.3/arch/arm/mach-mx28/mx28evk_pins.c
+++ b/linux-2.6.35.3/arch/arm/mach-mx28/mx28evk_pins.c
@@ -950,8 +950,8 @@ static struct pin_desc mx28evk_gpmi_pins[] = {
 	 },
 };
 
-static struct pin_desc mx28evk_atheros_pins[] = {
-	/* SPI interface */
+/* QCA7000 SPI pins */
+static struct pin_desc mx28evk_atheros_pins_spi[] = {
 	{
 	 .name	= "SSP2 MOSI",
 	 .id	= PINID_SSP2_MOSI,
@@ -985,6 +985,57 @@ static struct pin_desc mx28evk_atheros_pins[] = {
 	 .drive 	= 1,
 	 },
 
+	/* QCA7000 Reset */
+	{
+	 .name  = "QCA7K_RESET",
+	 .id	= PINID_LCD_D13,
+	 .fun	= PIN_GPIO,
+	 .strength = PAD_4MA,
+	 .voltage = PAD_3_3V,
+	 .drive	= 1,
+	 .output = 0,
+	 .data = 1,
+	 },
+};
+
+/* QCA7000 UART pins */
+static struct pin_desc mx28evk_atheros_pins_uart[] = {
+	{
+	 .name  = "AUART0.RX",
+	 .id    = PINID_AUART0_RX,
+	 .fun   = PIN_FUN1,
+	 .strength = PAD_4MA,
+	 .voltage = PAD_3_3V,
+	 .drive = 1,
+	 },
+	{
+	 .name  = "AUART0.TX",
+	 .id    = PINID_AUART0_TX,
+	 .fun   = PIN_FUN1,
+	 .strength = PAD_4MA,
+	 .voltage = PAD_3_3V,
+	 .drive = 1,
+	 },
+	{
+	 .name  = "AUART0.CTS",
+	 .id    = PINID_AUART0_CTS,
+	 .fun   = PIN_FUN1,
+	 .strength = PAD_4MA,
+	 .voltage = PAD_3_3V,
+	 .drive = 1,
+	 },
+	{
+	 .name  = "AUART0.RTS",
+	 .id    = PINID_AUART0_RTS,
+	 .fun   = PIN_FUN1,
+	 .strength = PAD_4MA,
+	 .voltage = PAD_3_3V,
+	 .drive = 1,
+	 },
+};
+
+/* Input pins (SPI interrupts, SW13, GPIOs) */
+static struct pin_desc mx28evk_atheros_pins[] = {
 	/* QCA7000 interrupt lines */
 	{
 	 .name = "QCA7K_INT0",
@@ -1014,17 +1065,6 @@ static struct pin_desc mx28evk_atheros_pins[] = {
 	 .output = 0,
 	 },
 
-	/* QCA7000 Reset */
-	{
-	 .name  = "QCA7K_RESET",
-	 .id	= PINID_LCD_D13,
-	 .fun	= PIN_GPIO,
-	 .strength = PAD_4MA,
-	 .voltage = PAD_3_3V,
-	 .drive	= 1,
-	 .output = 0,
-	 },
-
 	/* QCA7000 GPIOs */
 	{
 	 .name = "QCA7K_GPIO_0",
@@ -1037,7 +1077,7 @@ static struct pin_desc mx28evk_atheros_pins[] = {
 	 },
 	{
 	 .name = "QCA7K_GPIO_1",
-	 .id   = PINID_LCD_VSYNC,
+	 .id   = PINID_LCD_HSYNC,
 	 .fun  = PIN_GPIO,
 	 .strength = PAD_4MA,
 	 .voltage = PAD_3_3V,
@@ -1046,7 +1086,7 @@ static struct pin_desc mx28evk_atheros_pins[] = {
 	 },
 	{
 	 .name = "QCA7K_GPIO_2",
-	 .id = PINID_LCD_HSYNC,
+	 .id = PINID_LCD_VSYNC,
 	 .fun = PIN_GPIO,
 	 .strength = PAD_4MA,
 	 .voltage = PAD_3_3V,
@@ -1063,79 +1103,6 @@ static struct pin_desc mx28evk_atheros_pins[] = {
 	 .output = 0,
 	 },
 
-	 /* QCA7000 UART interface */
-#if 0
-	{
-	 .name  = "AUART0.RX",
-	 .id    = PINID_AUART0_RX,
-	 .fun   = PIN_FUN1,
-	 .strength = PAD_4MA,
-	 .voltage = PAD_3_3V,
-	 .drive = 1,
-	 },
-	{
-	 .name  = "AUART0.TX",
-	 .id    = PINID_AUART0_TX,
-	 .fun   = PIN_FUN1,
-	 .strength = PAD_4MA,
-	 .voltage = PAD_3_3V,
-	 .drive = 1,
-	 },
-	{
-	 .name  = "AUART0.CTS",
-	 .id    = PINID_AUART0_CTS,
-	 .fun   = PIN_FUN1,
-	 .strength = PAD_4MA,
-	 .voltage = PAD_3_3V,
-	 .drive = 1,
-	 },
-	{
-	 .name  = "AUART0.RTS",
-	 .id    = PINID_AUART0_RTS,
-	 .fun   = PIN_FUN1,
-	 .strength = PAD_4MA,
-	 .voltage = PAD_3_3V,
-	 .drive = 1,
-	 },
-#else
-	{
-	 .name  = "AUART0.RX",
-	 .id    = PINID_AUART0_RX,
-	 .fun   = PIN_GPIO,
-	 .strength = PAD_4MA,
-	 .voltage = PAD_3_3V,
-	 .drive = 1,
-	 .output = 0
-	 },
-	{
-	 .name  = "AUART0.TX",
-	 .id    = PINID_AUART0_TX,
-	 .fun   = PIN_GPIO,
-	 .strength = PAD_4MA,
-	 .voltage = PAD_3_3V,
-	 .drive = 1,
-	 .output = 0
-	 },
-	{
-	 .name  = "AUART0.CTS",
-	 .id    = PINID_AUART0_CTS,
-	 .fun   = PIN_GPIO,
-	 .strength = PAD_4MA,
-	 .voltage = PAD_3_3V,
-	 .drive = 1,
-	 .output = 0
-	 },
-	{
-	 .name  = "AUART0.RTS",
-	 .id    = PINID_AUART0_RTS,
-	 .fun   = PIN_GPIO,
-	 .strength = PAD_4MA,
-	 .voltage = PAD_3_3V,
-	 .drive = 1,
-	 .output = 0
-	 },
-#endif
-
 	/* Configuration SW13 */
 	{
 	 .name  = "SW13 0",
@@ -1268,9 +1235,69 @@ void __init mx28evk_init_pin_group(struct pin_desc *pins, unsigned count)
 	}
 }
 
-void __init mx28evk_pins_init(void)
+#define ATHEROS_PIN_SPI_UART_SELECT    (PINID_LCD_D12)
+#define ATHEROS_PIN_RS485_RS232_SELECT (PINID_LCD_D11)
+#define ATHEROS_PIN_XCVR_ENABLED       (PINID_LCD_D10)
+#define ATHEROS_PIN_DISABLE_OUTPUTS    (PINID_LCD_D07)
+
+#define TEST_READ_GPIO(x, y) { uint32_t temp_value = gpio_get_value(MXS_PIN_TO_GPIO(x)); printk(KERN_DEBUG "QCA ===> %s (" #x "): 0x%08x\n", y, temp_value); }
+
+void
+atheros_pin_cfg(void)
 {
+	uint32_t rs232_enabled;
+	uint32_t disable_outputs;
+	uint32_t spi_mode;
+	uint32_t xcvr_enabled;
+
+	/* Setup input pins. */
+	mx28evk_init_pin_group(mx28evk_atheros_pins,
+	    ARRAY_SIZE(mx28evk_atheros_pins));
+
+#if 1
+	/* attempt to read switch configuration */
+	TEST_READ_GPIO(PINID_LCD_D12, "DIP1");
+	TEST_READ_GPIO(PINID_LCD_D11, "DIP2");
+	TEST_READ_GPIO(PINID_LCD_D10, "DIP3");
+	TEST_READ_GPIO(PINID_LCD_D07, "DIP4");
 
+	TEST_READ_GPIO(PINID_LCD_D13, "RESET OUT");
+
+	TEST_READ_GPIO(PINID_LCD_WR_RWN, "GPIO 0");
+	TEST_READ_GPIO(PINID_LCD_HSYNC, "GPIO 1");
+	TEST_READ_GPIO(PINID_LCD_VSYNC, "GPIO 2");
+	TEST_READ_GPIO(PINID_LCD_DOTCK, "GPIO 3");
+#endif
+
+	rs232_enabled = !gpio_get_value(MXS_PIN_TO_GPIO(ATHEROS_PIN_RS485_RS232_SELECT));
+	disable_outputs = gpio_get_value(MXS_PIN_TO_GPIO(ATHEROS_PIN_DISABLE_OUTPUTS));
+	spi_mode = !gpio_get_value(MXS_PIN_TO_GPIO(ATHEROS_PIN_SPI_UART_SELECT));
+	xcvr_enabled = gpio_get_value(MXS_PIN_TO_GPIO(ATHEROS_PIN_XCVR_ENABLED));
+
+	printk(KERN_DEBUG "QCA ===> SW13: (rs232 enable: %d, disable_outputs: %d, xcvr_enabled: %d)\n", rs232_enabled != 0, disable_outputs != 0, xcvr_enabled != 0);
+
+	/* leave outputs disabled */
+	if (rs232_enabled || disable_outputs || xcvr_enabled) {
+		printk(KERN_DEBUG "QCA ===> Leaving outputs disabled.\n");
+		return;
+	}
+
+	/*
+	 * Otherwise, setup UART or SPI mode.
+	 */
+	if (spi_mode) {
+		printk(KERN_DEBUG "QCA ===> Enabling SPI pins\n");
+		mx28evk_init_pin_group(mx28evk_atheros_pins_spi,
+		    ARRAY_SIZE(mx28evk_atheros_pins_spi));
+	} else {
+		printk(KERN_DEBUG "QCA ===> Enabling UART pins\n");
+		mx28evk_init_pin_group(mx28evk_atheros_pins_uart,
+		    ARRAY_SIZE(mx28evk_atheros_pins_uart));
+	}
+}
+
+void __init mx28evk_pins_init(void)
+{
 	mx28evk_init_pin_group(mx28evk_fixed_pins,
 						ARRAY_SIZE(mx28evk_fixed_pins));
 
@@ -1284,8 +1311,8 @@ void __init mx28evk_pins_init(void)
 						ARRAY_SIZE(mx28evk_gpmi_pins));
 	}
 
-	mx28evk_init_pin_group(mx28evk_atheros_pins,
-					ARRAY_SIZE(mx28evk_atheros_pins));
+	atheros_pin_cfg();
+
 
 #if defined(CONFIG_FEC) || defined(CONFIG_FEC_MODULE)\
 	|| defined(CONFIG_FEC_L2SWITCH)
@@ -1300,9 +1327,7 @@ void __init mx28evk_pins_init(void)
 
 int gpio_spi_intr_cfg(void)
 {
-	set_irq_type(gpio_to_irq(MXS_PIN_TO_GPIO(PINID_I2C0_SDA)), IRQ_TYPE_EDGE_RISING);
-
-	return 0;
+	return set_irq_type(gpio_to_irq(MXS_PIN_TO_GPIO(PINID_I2C0_SDA)), IRQ_TYPE_EDGE_RISING);
 }
 
 int gpio_spi_intr_get_irq(void)
-- 
1.7.4.1


From 0cd907124b6e0f0dea6bd7f6b7593ce4bfeb4b55 Mon Sep 17 00:00:00 2001
From: Nathaniel Houghton <nhoughto@qca.qualcomm.com>
Date: Tue, 3 Jan 2012 09:50:57 -0500
Subject: [PATCH 11/15] add N_QCA line discipline;
 update slattach to support it;

---
 linux-2.6.35.3/include/linux/tty.h |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/linux-2.6.35.3/include/linux/tty.h b/linux-2.6.35.3/include/linux/tty.h
index 931078b..91efcf9 100644
--- a/linux-2.6.35.3/include/linux/tty.h
+++ b/linux-2.6.35.3/include/linux/tty.h
@@ -49,6 +49,7 @@
 #define N_V253		19	/* Codec control over voice modem */
 #define N_CAIF		20      /* CAIF protocol for talking to modems */
 #define N_GSM0710	21	/* GSM 0710 Mux */
+#define N_QCA		22	/* QCA UART Protocol */
 
 /*
  * This character is the same as _POSIX_VDISABLE: it cannot be used as
-- 
1.7.4.1


From 3e7aa3e7be8fe4ef7cc0e52e867d0e21e9e865a3 Mon Sep 17 00:00:00 2001
From: Nathaniel Houghton <nhoughto@qca.qualcomm.com>
Date: Wed, 4 Jan 2012 18:38:08 -0500
Subject: [PATCH 12/15] add support for reading bootstrap options;
 add qca_bootstrap module;

---
 linux-2.6.35.3/arch/arm/mach-mx28/mx28evk_pins.c   |   83 +++++++++++---------
 .../arch/arm/plat-mxs/include/mach/qca_bootstrap.h |   18 ++++
 2 files changed, 65 insertions(+), 36 deletions(-)
 create mode 100644 linux-2.6.35.3/arch/arm/plat-mxs/include/mach/qca_bootstrap.h

diff --git a/linux-2.6.35.3/arch/arm/mach-mx28/mx28evk_pins.c b/linux-2.6.35.3/arch/arm/mach-mx28/mx28evk_pins.c
index 1476768..03aecf6 100644
--- a/linux-2.6.35.3/arch/arm/mach-mx28/mx28evk_pins.c
+++ b/linux-2.6.35.3/arch/arm/mach-mx28/mx28evk_pins.c
@@ -24,6 +24,7 @@
 #include <linux/delay.h>
 
 #include <mach/pinctrl.h>
+#include <mach/qca_bootstrap.h>
 
 #include "mx28_pins.h"
 
@@ -951,7 +952,7 @@ static struct pin_desc mx28evk_gpmi_pins[] = {
 };
 
 /* QCA7000 SPI pins */
-static struct pin_desc mx28evk_atheros_pins_spi[] = {
+static struct pin_desc mx28evk_qca_pins_spi[] = {
 	{
 	 .name	= "SSP2 MOSI",
 	 .id	= PINID_SSP2_MOSI,
@@ -999,7 +1000,7 @@ static struct pin_desc mx28evk_atheros_pins_spi[] = {
 };
 
 /* QCA7000 UART pins */
-static struct pin_desc mx28evk_atheros_pins_uart[] = {
+static struct pin_desc mx28evk_qca_pins_uart[] = {
 	{
 	 .name  = "AUART0.RX",
 	 .id    = PINID_AUART0_RX,
@@ -1035,7 +1036,7 @@ static struct pin_desc mx28evk_atheros_pins_uart[] = {
 };
 
 /* Input pins (SPI interrupts, SW13, GPIOs) */
-static struct pin_desc mx28evk_atheros_pins[] = {
+static struct pin_desc mx28evk_qca_pins[] = {
 	/* QCA7000 interrupt lines */
 	{
 	 .name = "QCA7K_INT0",
@@ -1235,15 +1236,15 @@ void __init mx28evk_init_pin_group(struct pin_desc *pins, unsigned count)
 	}
 }
 
-#define ATHEROS_PIN_SPI_UART_SELECT    (PINID_LCD_D12)
-#define ATHEROS_PIN_RS485_RS232_SELECT (PINID_LCD_D11)
-#define ATHEROS_PIN_XCVR_ENABLED       (PINID_LCD_D10)
-#define ATHEROS_PIN_DISABLE_OUTPUTS    (PINID_LCD_D07)
+#define QCA_PIN_SPI_UART_SELECT    (PINID_LCD_D12)
+#define QCA_PIN_RS232_RS485_SELECT (PINID_LCD_D11)
+#define QCA_PIN_XCVR_ENABLED       (PINID_LCD_D10)
+#define QCA_PIN_DISABLE_OUTPUTS    (PINID_LCD_D07)
 
-#define TEST_READ_GPIO(x, y) { uint32_t temp_value = gpio_get_value(MXS_PIN_TO_GPIO(x)); printk(KERN_DEBUG "QCA ===> %s (" #x "): 0x%08x\n", y, temp_value); }
+static uint32_t qca_bootstraps;
 
 void
-atheros_pin_cfg(void)
+qca_pin_cfg(void)
 {
 	uint32_t rs232_enabled;
 	uint32_t disable_outputs;
@@ -1251,33 +1252,32 @@ atheros_pin_cfg(void)
 	uint32_t xcvr_enabled;
 
 	/* Setup input pins. */
-	mx28evk_init_pin_group(mx28evk_atheros_pins,
-	    ARRAY_SIZE(mx28evk_atheros_pins));
+	mx28evk_init_pin_group(mx28evk_qca_pins,
+	    ARRAY_SIZE(mx28evk_qca_pins));
 
-#if 1
-	/* attempt to read switch configuration */
-	TEST_READ_GPIO(PINID_LCD_D12, "DIP1");
-	TEST_READ_GPIO(PINID_LCD_D11, "DIP2");
-	TEST_READ_GPIO(PINID_LCD_D10, "DIP3");
-	TEST_READ_GPIO(PINID_LCD_D07, "DIP4");
-
-	TEST_READ_GPIO(PINID_LCD_D13, "RESET OUT");
-
-	TEST_READ_GPIO(PINID_LCD_WR_RWN, "GPIO 0");
-	TEST_READ_GPIO(PINID_LCD_HSYNC, "GPIO 1");
-	TEST_READ_GPIO(PINID_LCD_VSYNC, "GPIO 2");
-	TEST_READ_GPIO(PINID_LCD_DOTCK, "GPIO 3");
-#endif
+	if (gpio_get_value(MXS_PIN_TO_GPIO(QCA_PIN_SPI_UART_SELECT))) {
+		qca_bootstraps |= QCA_SPI_UART_SELECT;
+	}
+	if (gpio_get_value(MXS_PIN_TO_GPIO(QCA_PIN_RS232_RS485_SELECT))) {
+		qca_bootstraps |= QCA_RS232_RS485_SELECT;
+	}
+	if (gpio_get_value(MXS_PIN_TO_GPIO(QCA_PIN_XCVR_ENABLED))) {
+		qca_bootstraps |= QCA_XCVR_ENABLED;
+	}
+	if (gpio_get_value(MXS_PIN_TO_GPIO(QCA_PIN_DISABLE_OUTPUTS))) {
+		qca_bootstraps |= QCA_DISABLE_OUTPUTS;
+	}
 
-	rs232_enabled = !gpio_get_value(MXS_PIN_TO_GPIO(ATHEROS_PIN_RS485_RS232_SELECT));
-	disable_outputs = gpio_get_value(MXS_PIN_TO_GPIO(ATHEROS_PIN_DISABLE_OUTPUTS));
-	spi_mode = !gpio_get_value(MXS_PIN_TO_GPIO(ATHEROS_PIN_SPI_UART_SELECT));
-	xcvr_enabled = gpio_get_value(MXS_PIN_TO_GPIO(ATHEROS_PIN_XCVR_ENABLED));
+	spi_mode = !gpio_get_value(MXS_PIN_TO_GPIO(QCA_PIN_SPI_UART_SELECT));
+	rs232_enabled = !gpio_get_value(MXS_PIN_TO_GPIO(QCA_PIN_RS232_RS485_SELECT));
+	xcvr_enabled = gpio_get_value(MXS_PIN_TO_GPIO(QCA_PIN_XCVR_ENABLED));
+	disable_outputs = gpio_get_value(MXS_PIN_TO_GPIO(QCA_PIN_DISABLE_OUTPUTS));
 
 	printk(KERN_DEBUG "QCA ===> SW13: (rs232 enable: %d, disable_outputs: %d, xcvr_enabled: %d)\n", rs232_enabled != 0, disable_outputs != 0, xcvr_enabled != 0);
 
 	/* leave outputs disabled */
 	if (rs232_enabled || disable_outputs || xcvr_enabled) {
+		qca_bootstraps |= QCA_HOST_MODE_DISABLED << QCA_HOST_MODE_SHIFT;
 		printk(KERN_DEBUG "QCA ===> Leaving outputs disabled.\n");
 		return;
 	}
@@ -1286,13 +1286,15 @@ atheros_pin_cfg(void)
 	 * Otherwise, setup UART or SPI mode.
 	 */
 	if (spi_mode) {
+		qca_bootstraps |= QCA_HOST_MODE_SPI << QCA_HOST_MODE_SHIFT;
 		printk(KERN_DEBUG "QCA ===> Enabling SPI pins\n");
-		mx28evk_init_pin_group(mx28evk_atheros_pins_spi,
-		    ARRAY_SIZE(mx28evk_atheros_pins_spi));
+		mx28evk_init_pin_group(mx28evk_qca_pins_spi,
+		    ARRAY_SIZE(mx28evk_qca_pins_spi));
 	} else {
+		qca_bootstraps |= QCA_HOST_MODE_UART << QCA_HOST_MODE_SHIFT;
 		printk(KERN_DEBUG "QCA ===> Enabling UART pins\n");
-		mx28evk_init_pin_group(mx28evk_atheros_pins_uart,
-		    ARRAY_SIZE(mx28evk_atheros_pins_uart));
+		mx28evk_init_pin_group(mx28evk_qca_pins_uart,
+		    ARRAY_SIZE(mx28evk_qca_pins_uart));
 	}
 }
 
@@ -1311,7 +1313,7 @@ void __init mx28evk_pins_init(void)
 						ARRAY_SIZE(mx28evk_gpmi_pins));
 	}
 
-	atheros_pin_cfg();
+	qca_pin_cfg();
 
 
 #if defined(CONFIG_FEC) || defined(CONFIG_FEC_MODULE)\
@@ -1325,15 +1327,24 @@ void __init mx28evk_pins_init(void)
  * Qualcomm Atheros SPI interrupt configuration functions.
  */
 
-int gpio_spi_intr_cfg(void)
+int
+gpio_spi_intr_cfg(void)
 {
 	return set_irq_type(gpio_to_irq(MXS_PIN_TO_GPIO(PINID_I2C0_SDA)), IRQ_TYPE_EDGE_RISING);
 }
 
-int gpio_spi_intr_get_irq(void)
+int
+gpio_spi_intr_get_irq(void)
 {
 	return gpio_to_irq(MXS_PIN_TO_GPIO(PINID_I2C0_SDA));
 }
 
+uint32_t
+qca_read_bootstrap(void)
+{
+	return qca_bootstraps;
+}
+
 EXPORT_SYMBOL(gpio_spi_intr_cfg);
 EXPORT_SYMBOL(gpio_spi_intr_get_irq);
+EXPORT_SYMBOL(qca_read_bootstrap);
diff --git a/linux-2.6.35.3/arch/arm/plat-mxs/include/mach/qca_bootstrap.h b/linux-2.6.35.3/arch/arm/plat-mxs/include/mach/qca_bootstrap.h
new file mode 100644
index 0000000..ee38ab8
--- /dev/null
+++ b/linux-2.6.35.3/arch/arm/plat-mxs/include/mach/qca_bootstrap.h
@@ -0,0 +1,18 @@
+#ifndef __ASM_ARCH_QCA_BOOTSTRAP_H__
+#define __ASM_ARCH_QCA_BOOTSTRAP_H__
+
+#define QCA_SPI_UART_SELECT    (1 << 0)
+#define QCA_RS232_RS485_SELECT (1 << 1)
+#define QCA_XCVR_ENABLED       (1 << 2)
+#define QCA_DISABLE_OUTPUTS    (1 << 3)
+
+#define QCA_HOST_MODE_SHIFT 4
+#define QCA_HOST_MODE (0x3 << QCA_HOST_MODE_SHIFT)
+
+#define QCA_HOST_MODE_DISABLED 0
+#define QCA_HOST_MODE_UART 1
+#define QCA_HOST_MODE_SPI 2
+
+uint32_t qca_read_bootstrap(void);
+
+#endif
-- 
1.7.4.1


From cde4c890b8e815e2cf43ba71f2c891eba4ca88ac Mon Sep 17 00:00:00 2001
From: Nathaniel Houghton <nhoughto@qca.qualcomm.com>
Date: Wed, 11 Jan 2012 18:13:55 -0500
Subject: [PATCH 13/15] add a function to reset the PHY;

reset the PHY before connecting to it in the fec driver. Works around
a PHY issue where the PHY is transmitting bogus data to the MAC;
---
 linux-2.6.35.3/arch/arm/mach-mx28/mx28evk_pins.c |   13 +++++++++++++
 linux-2.6.35.3/drivers/net/fec.c                 |    6 ++++++
 2 files changed, 19 insertions(+), 0 deletions(-)

diff --git a/linux-2.6.35.3/arch/arm/mach-mx28/mx28evk_pins.c b/linux-2.6.35.3/arch/arm/mach-mx28/mx28evk_pins.c
index 03aecf6..2adb712 100644
--- a/linux-2.6.35.3/arch/arm/mach-mx28/mx28evk_pins.c
+++ b/linux-2.6.35.3/arch/arm/mach-mx28/mx28evk_pins.c
@@ -1199,6 +1199,16 @@ void mx28evk_enet_io_lowerpower_exit(void)
 	}
 }
 
+void
+mx28evk_enet_phy_reset(void)
+{
+	/* reset phy */
+	gpio_direction_output(MXS_PIN_TO_GPIO(PINID_ENET0_RX_CLK), 0);
+	mdelay(50);
+	gpio_direction_output(MXS_PIN_TO_GPIO(PINID_ENET0_RX_CLK), 1);
+}
+EXPORT_SYMBOL(mx28evk_enet_phy_reset);
+
 #else
 int mx28evk_enet_gpio_init(void)
 {
@@ -1208,6 +1218,9 @@ void mx28evk_enet_io_lowerpower_enter(void)
 {}
 void mx28evk_enet_io_lowerpower_exit(void)
 {}
+void mx28evk_enet_phy_reset(void)
+{}
+EXPORT_SYMBOL(mx28evk_enet_phy_reset);
 #endif
 
 void __init mx28evk_init_pin_group(struct pin_desc *pins, unsigned count)
diff --git a/linux-2.6.35.3/drivers/net/fec.c b/linux-2.6.35.3/drivers/net/fec.c
index b980d70..0838be9 100644
--- a/linux-2.6.35.3/drivers/net/fec.c
+++ b/linux-2.6.35.3/drivers/net/fec.c
@@ -783,6 +783,8 @@ static int fec_enet_mdio_reset(struct mii_bus *bus)
 	return 0;
 }
 
+extern void mx28evk_enet_phy_reset(void);
+
 static int fec_enet_mii_probe(struct net_device *dev)
 {
 	struct fec_enet_private *fep = netdev_priv(dev);
@@ -807,6 +809,10 @@ static int fec_enet_mii_probe(struct net_device *dev)
 		return -ENODEV;
 	}
 
+	/* reset the PHY */
+	mx28evk_enet_phy_reset();
+	mdelay(50);
+
 	/* attach the mac to the phy */
 	phy_dev = phy_connect(dev, dev_name(&phy_dev->dev),
 			     &fec_enet_adjust_link, 0,
-- 
1.7.4.1


From ddcb395573c6546e6ff24f85c98d395abd2a4a4e Mon Sep 17 00:00:00 2001
From: Nathaniel Houghton <nhoughto@qca.qualcomm.com>
Date: Mon, 16 Jan 2012 11:49:50 -0500
Subject: [PATCH 14/15] don't return RTS status;

---
 linux-2.6.35.3/drivers/serial/mxs-auart.c |    4 ----
 1 files changed, 0 insertions(+), 4 deletions(-)

diff --git a/linux-2.6.35.3/drivers/serial/mxs-auart.c b/linux-2.6.35.3/drivers/serial/mxs-auart.c
index be79f0e..9b66aa5 100644
--- a/linux-2.6.35.3/drivers/serial/mxs-auart.c
+++ b/linux-2.6.35.3/drivers/serial/mxs-auart.c
@@ -465,16 +465,12 @@ static u32 mxs_auart_get_mctrl(struct uart_port *u)
 {
 	struct mxs_auart_port *s = to_auart_port(u);
 	u32 stat = __raw_readl(u->membase + HW_UARTAPP_STAT);
-	int ctrl2 = __raw_readl(u->membase + HW_UARTAPP_CTRL2);
 	u32 mctrl = s->ctrl;
 
 	mctrl &= ~TIOCM_CTS;
 	if (stat & BM_UARTAPP_STAT_CTS)
 		mctrl |= TIOCM_CTS;
 
-	if (ctrl2 & BM_UARTAPP_CTRL2_RTS)
-		mctrl |= TIOCM_RTS;
-
 	return mctrl;
 }
 
-- 
1.7.4.1


From a3319681ee4c650804bf11c25465e85d0da78a5d Mon Sep 17 00:00:00 2001
From: Nathaniel Houghton <nhoughto@qca.qualcomm.com>
Date: Thu, 19 Jan 2012 16:00:15 -0500
Subject: [PATCH 15/15] update default MX28 config;

---
 linux-2.6.35.3/arch/arm/configs/imx28evk_defconfig |  913 ++------------------
 1 files changed, 68 insertions(+), 845 deletions(-)

diff --git a/linux-2.6.35.3/arch/arm/configs/imx28evk_defconfig b/linux-2.6.35.3/arch/arm/configs/imx28evk_defconfig
index 0d01c03..b7b9ffa 100644
--- a/linux-2.6.35.3/arch/arm/configs/imx28evk_defconfig
+++ b/linux-2.6.35.3/arch/arm/configs/imx28evk_defconfig
@@ -1,7 +1,6 @@
 #
 # Automatically generated make config: don't edit
 # Linux kernel version: 2.6.35.3
-# Mon Dec  6 17:17:47 2010
 #
 CONFIG_ARM=y
 CONFIG_SYS_SUPPORTS_APM_EMULATION=y
@@ -465,10 +464,13 @@ CONFIG_NETWORK_SECMARK=y
 # CONFIG_TIPC is not set
 # CONFIG_ATM is not set
 # CONFIG_L2TP is not set
-# CONFIG_BRIDGE is not set
+CONFIG_STP=y
+CONFIG_BRIDGE=y
+CONFIG_BRIDGE_IGMP_SNOOPING=y
 # CONFIG_NET_DSA is not set
 # CONFIG_VLAN_8021Q is not set
 # CONFIG_DECNET is not set
+CONFIG_LLC=y
 # CONFIG_LLC2 is not set
 # CONFIG_IPX is not set
 # CONFIG_ATALK is not set
@@ -487,40 +489,13 @@ CONFIG_NETWORK_SECMARK=y
 # CONFIG_NET_PKTGEN is not set
 # CONFIG_NET_TCPPROBE is not set
 # CONFIG_HAMRADIO is not set
-CONFIG_CAN=y
-CONFIG_CAN_RAW=y
-# CONFIG_CAN_BCM is not set
-
-#
-# CAN Device Drivers
-#
-# CONFIG_CAN_VCAN is not set
-# CONFIG_CAN_DEV is not set
-# CONFIG_CAN_DEBUG_DEVICES is not set
-CONFIG_CAN_FLEXCAN=m
+# CONFIG_CAN is not set
 # CONFIG_IRDA is not set
 # CONFIG_BT is not set
 # CONFIG_AF_RXRPC is not set
-CONFIG_WIRELESS=y
-CONFIG_WIRELESS_EXT=y
-CONFIG_WEXT_CORE=y
-CONFIG_WEXT_PROC=y
-CONFIG_WEXT_SPY=y
-CONFIG_WEXT_PRIV=y
-# CONFIG_CFG80211 is not set
-CONFIG_WIRELESS_EXT_SYSFS=y
-# CONFIG_LIB80211 is not set
-
-#
-# CFG80211 needs to be enabled for MAC80211
-#
-
-#
-# Some wireless drivers require a rate control algorithm
-#
+# CONFIG_WIRELESS is not set
 # CONFIG_WIMAX is not set
-CONFIG_RFKILL=y
-CONFIG_RFKILL_INPUT=y
+# CONFIG_RFKILL is not set
 # CONFIG_NET_9P is not set
 # CONFIG_CAIF is not set
 
@@ -594,8 +569,7 @@ CONFIG_MTD_CFI_I2=y
 #
 # CONFIG_MTD_DATAFLASH is not set
 # CONFIG_MTD_MXC_DATAFLASH is not set
-CONFIG_MTD_M25P80=m
-CONFIG_M25PXX_USE_FAST_READ=y
+# CONFIG_MTD_M25P80 is not set
 # CONFIG_MTD_SST25L is not set
 # CONFIG_MTD_SLRAM is not set
 # CONFIG_MTD_PHRAM is not set
@@ -621,7 +595,6 @@ CONFIG_MTD_NAND_IDS=y
 # CONFIG_MTD_NAND_NANDSIM is not set
 CONFIG_MTD_NAND_GPMI_NFC=y
 # CONFIG_MTD_NAND_PLATFORM is not set
-# CONFIG_MTD_ALAUDA is not set
 # CONFIG_MTD_ONENAND is not set
 
 #
@@ -652,7 +625,6 @@ CONFIG_BLK_DEV_LOOP=y
 # DRBD disabled because PROC_FS, INET or CONNECTOR not selected
 #
 # CONFIG_BLK_DEV_NBD is not set
-# CONFIG_BLK_DEV_UB is not set
 CONFIG_BLK_DEV_RAM=y
 CONFIG_BLK_DEV_RAM_COUNT=4
 CONFIG_BLK_DEV_RAM_SIZE=16384
@@ -662,11 +634,7 @@ CONFIG_BLK_DEV_RAM_SIZE=16384
 # CONFIG_MG_DISK is not set
 CONFIG_MISC_DEVICES=y
 # CONFIG_AD525X_DPOT is not set
-# CONFIG_ICS932S401 is not set
 # CONFIG_ENCLOSURE_SERVICES is not set
-# CONFIG_ISL29003 is not set
-# CONFIG_SENSORS_TSL2550 is not set
-# CONFIG_DS1682 is not set
 # CONFIG_TI_DAC7512 is not set
 CONFIG_MXS_PERSISTENT=y
 # CONFIG_ANDROID_PMEM is not set
@@ -676,10 +644,7 @@ CONFIG_MXS_PERSISTENT=y
 #
 # EEPROM support
 #
-# CONFIG_EEPROM_AT24 is not set
 # CONFIG_EEPROM_AT25 is not set
-# CONFIG_EEPROM_LEGACY is not set
-# CONFIG_EEPROM_MAX6875 is not set
 # CONFIG_EEPROM_93CX6 is not set
 # CONFIG_IWMC3200TOP is not set
 CONFIG_HAVE_IDE=y
@@ -690,43 +655,9 @@ CONFIG_HAVE_IDE=y
 #
 CONFIG_SCSI_MOD=y
 # CONFIG_RAID_ATTRS is not set
-CONFIG_SCSI=y
-CONFIG_SCSI_DMA=y
-# CONFIG_SCSI_TGT is not set
-CONFIG_SCSI_NETLINK=y
-CONFIG_SCSI_PROC_FS=y
-
-#
-# SCSI support type (disk, tape, CD-ROM)
-#
-CONFIG_BLK_DEV_SD=y
-# CONFIG_CHR_DEV_ST is not set
-# CONFIG_CHR_DEV_OSST is not set
-# CONFIG_BLK_DEV_SR is not set
-# CONFIG_CHR_DEV_SG is not set
-# CONFIG_CHR_DEV_SCH is not set
-CONFIG_SCSI_MULTI_LUN=y
-# CONFIG_SCSI_CONSTANTS is not set
-# CONFIG_SCSI_LOGGING is not set
-# CONFIG_SCSI_SCAN_ASYNC is not set
-CONFIG_SCSI_WAIT_SCAN=m
-
-#
-# SCSI Transports
-#
-CONFIG_SCSI_SPI_ATTRS=m
-CONFIG_SCSI_FC_ATTRS=m
-CONFIG_SCSI_ISCSI_ATTRS=m
-# CONFIG_SCSI_SAS_ATTRS is not set
-# CONFIG_SCSI_SAS_LIBSAS is not set
-# CONFIG_SCSI_SRP_ATTRS is not set
-CONFIG_SCSI_LOWLEVEL=y
-# CONFIG_ISCSI_TCP is not set
-# CONFIG_LIBFC is not set
-# CONFIG_LIBFCOE is not set
-# CONFIG_SCSI_DEBUG is not set
-# CONFIG_SCSI_DH is not set
-# CONFIG_SCSI_OSD_INITIATOR is not set
+# CONFIG_SCSI is not set
+# CONFIG_SCSI_DMA is not set
+# CONFIG_SCSI_NETLINK is not set
 # CONFIG_ATA is not set
 # CONFIG_MD is not set
 CONFIG_NETDEVICES=y
@@ -783,35 +714,11 @@ CONFIG_FEC=y
 # CONFIG_FEC2 is not set
 # CONFIG_NETDEV_1000 is not set
 # CONFIG_NETDEV_10000 is not set
-CONFIG_WLAN=y
-# CONFIG_USB_ZD1201 is not set
-# CONFIG_HOSTAP is not set
-CONFIG_ATH6K_LEGACY=m
-CONFIG_AR600x_SD31_XXX=y
-# CONFIG_AR600x_WB31_XXX is not set
-# CONFIG_AR600x_SD32_XXX is not set
-# CONFIG_AR600x_CUSTOM_XXX is not set
-# CONFIG_ATH6KL_ENABLE_COEXISTENCE is not set
-# CONFIG_ATH6KL_HCI_BRIDGE is not set
-# CONFIG_ATH6KL_CFG80211 is not set
-# CONFIG_ATH6KL_HTC_RAW_INTERFACE is not set
-# CONFIG_ATH6KL_VIRTUAL_SCATTER_GATHER is not set
-# CONFIG_ATH6KL_DEBUG is not set
+# CONFIG_WLAN is not set
 
 #
 # Enable WiMAX (Networking options) to see the WiMAX drivers
 #
-
-#
-# USB Network Adapters
-#
-# CONFIG_USB_CATC is not set
-# CONFIG_USB_KAWETH is not set
-# CONFIG_USB_PEGASUS is not set
-# CONFIG_USB_RTL8150 is not set
-# CONFIG_USB_USBNET is not set
-# CONFIG_USB_HSO is not set
-# CONFIG_USB_IPHETH is not set
 # CONFIG_WAN is not set
 # CONFIG_PPP is not set
 # CONFIG_SLIP is not set
@@ -825,7 +732,7 @@ CONFIG_AR600x_SD31_XXX=y
 # Input device support
 #
 CONFIG_INPUT=y
-CONFIG_INPUT_FF_MEMLESS=y
+# CONFIG_INPUT_FF_MEMLESS is not set
 # CONFIG_INPUT_POLLDEV is not set
 # CONFIG_INPUT_SPARSEKMAP is not set
 
@@ -833,116 +740,27 @@ CONFIG_INPUT_FF_MEMLESS=y
 # Userland interfaces
 #
 CONFIG_INPUT_MOUSEDEV=y
-CONFIG_INPUT_MOUSEDEV_PSAUX=y
+# CONFIG_INPUT_MOUSEDEV_PSAUX is not set
 CONFIG_INPUT_MOUSEDEV_SCREEN_X=1024
 CONFIG_INPUT_MOUSEDEV_SCREEN_Y=768
-CONFIG_INPUT_JOYDEV=m
-CONFIG_INPUT_EVDEV=y
+# CONFIG_INPUT_JOYDEV is not set
+# CONFIG_INPUT_EVDEV is not set
 # CONFIG_INPUT_EVBUG is not set
 
 #
 # Input Device Drivers
 #
-CONFIG_INPUT_KEYBOARD=y
-# CONFIG_KEYBOARD_ADP5588 is not set
-CONFIG_KEYBOARD_ATKBD=y
-# CONFIG_KEYBOARD_QT2160 is not set
-# CONFIG_KEYBOARD_LKKBD is not set
-# CONFIG_KEYBOARD_GPIO is not set
-# CONFIG_KEYBOARD_TCA6416 is not set
-# CONFIG_KEYBOARD_MATRIX is not set
-# CONFIG_KEYBOARD_LM8323 is not set
-# CONFIG_KEYBOARD_MAX7359 is not set
-# CONFIG_KEYBOARD_NEWTON is not set
-# CONFIG_KEYBOARD_OPENCORES is not set
-# CONFIG_KEYBOARD_STOWAWAY is not set
-# CONFIG_KEYBOARD_SUNKBD is not set
-# CONFIG_KEYBOARD_XTKBD is not set
-CONFIG_KEYBOARD_MXS=y
-CONFIG_INPUT_MOUSE=y
-CONFIG_MOUSE_PS2=y
-CONFIG_MOUSE_PS2_ALPS=y
-CONFIG_MOUSE_PS2_LOGIPS2PP=y
-CONFIG_MOUSE_PS2_SYNAPTICS=y
-CONFIG_MOUSE_PS2_TRACKPOINT=y
-# CONFIG_MOUSE_PS2_ELANTECH is not set
-# CONFIG_MOUSE_PS2_SENTELIC is not set
-# CONFIG_MOUSE_PS2_TOUCHKIT is not set
-CONFIG_MOUSE_SERIAL=m
-# CONFIG_MOUSE_APPLETOUCH is not set
-# CONFIG_MOUSE_BCM5974 is not set
-CONFIG_MOUSE_VSXXXAA=m
-# CONFIG_MOUSE_GPIO is not set
-# CONFIG_MOUSE_SYNAPTICS_I2C is not set
-CONFIG_INPUT_JOYSTICK=y
-# CONFIG_JOYSTICK_ANALOG is not set
-# CONFIG_JOYSTICK_A3D is not set
-# CONFIG_JOYSTICK_ADI is not set
-# CONFIG_JOYSTICK_COBRA is not set
-# CONFIG_JOYSTICK_GF2K is not set
-# CONFIG_JOYSTICK_GRIP is not set
-# CONFIG_JOYSTICK_GRIP_MP is not set
-# CONFIG_JOYSTICK_GUILLEMOT is not set
-# CONFIG_JOYSTICK_INTERACT is not set
-# CONFIG_JOYSTICK_SIDEWINDER is not set
-# CONFIG_JOYSTICK_TMDC is not set
-# CONFIG_JOYSTICK_IFORCE is not set
-# CONFIG_JOYSTICK_WARRIOR is not set
-# CONFIG_JOYSTICK_MAGELLAN is not set
-# CONFIG_JOYSTICK_SPACEORB is not set
-# CONFIG_JOYSTICK_SPACEBALL is not set
-# CONFIG_JOYSTICK_STINGER is not set
-# CONFIG_JOYSTICK_TWIDJOY is not set
-# CONFIG_JOYSTICK_ZHENHUA is not set
-# CONFIG_JOYSTICK_JOYDUMP is not set
-# CONFIG_JOYSTICK_XPAD is not set
+# CONFIG_INPUT_KEYBOARD is not set
+# CONFIG_INPUT_MOUSE is not set
+# CONFIG_INPUT_JOYSTICK is not set
 # CONFIG_INPUT_TABLET is not set
-CONFIG_INPUT_TOUCHSCREEN=y
-# CONFIG_TOUCHSCREEN_ADS7846 is not set
-# CONFIG_TOUCHSCREEN_AD7877 is not set
-# CONFIG_TOUCHSCREEN_AD7879_I2C is not set
-# CONFIG_TOUCHSCREEN_AD7879_SPI is not set
-# CONFIG_TOUCHSCREEN_AD7879 is not set
-# CONFIG_TOUCHSCREEN_DYNAPRO is not set
-# CONFIG_TOUCHSCREEN_HAMPSHIRE is not set
-# CONFIG_TOUCHSCREEN_EETI is not set
-# CONFIG_TOUCHSCREEN_FUJITSU is not set
-CONFIG_TOUCHSCREEN_GUNZE=m
-# CONFIG_TOUCHSCREEN_ELO is not set
-# CONFIG_TOUCHSCREEN_WACOM_W8001 is not set
-# CONFIG_TOUCHSCREEN_MCS5000 is not set
-# CONFIG_TOUCHSCREEN_MTOUCH is not set
-# CONFIG_TOUCHSCREEN_INEXIO is not set
-# CONFIG_TOUCHSCREEN_MK712 is not set
-CONFIG_TOUCHSCREEN_MXS=y
-# CONFIG_TOUCHSCREEN_PENMOUNT is not set
-# CONFIG_TOUCHSCREEN_TOUCHRIGHT is not set
-# CONFIG_TOUCHSCREEN_TOUCHWIN is not set
-# CONFIG_TOUCHSCREEN_USB_COMPOSITE is not set
-# CONFIG_TOUCHSCREEN_TOUCHIT213 is not set
-# CONFIG_TOUCHSCREEN_TSC2007 is not set
-# CONFIG_TOUCHSCREEN_W90X900 is not set
-# CONFIG_TOUCHSCREEN_TPS6507X is not set
-CONFIG_INPUT_MISC=y
-# CONFIG_INPUT_AD714X is not set
-# CONFIG_INPUT_ATI_REMOTE is not set
-# CONFIG_INPUT_ATI_REMOTE2 is not set
-# CONFIG_INPUT_KEYSPAN_REMOTE is not set
-# CONFIG_INPUT_POWERMATE is not set
-# CONFIG_INPUT_YEALINK is not set
-# CONFIG_INPUT_CM109 is not set
-CONFIG_INPUT_UINPUT=m
-# CONFIG_INPUT_PCF8574 is not set
-# CONFIG_INPUT_GPIO_ROTARY_ENCODER is not set
+# CONFIG_INPUT_TOUCHSCREEN is not set
+# CONFIG_INPUT_MISC is not set
 
 #
 # Hardware I/O ports
 #
-CONFIG_SERIO=y
-CONFIG_SERIO_SERPORT=y
-CONFIG_SERIO_LIBPS2=y
-# CONFIG_SERIO_RAW is not set
-# CONFIG_SERIO_ALTERA_PS2 is not set
+# CONFIG_SERIO is not set
 # CONFIG_GAMEPORT is not set
 
 #
@@ -954,12 +772,8 @@ CONFIG_VT_CONSOLE=y
 CONFIG_HW_CONSOLE=y
 # CONFIG_VT_HW_CONSOLE_BINDING is not set
 CONFIG_DEVKMEM=y
-CONFIG_SERIAL_NONSTANDARD=y
-# CONFIG_N_HDLC is not set
+# CONFIG_SERIAL_NONSTANDARD is not set
 # CONFIG_N_GSM is not set
-# CONFIG_RISCOM8 is not set
-# CONFIG_SPECIALIX is not set
-CONFIG_STALDRV=y
 CONFIG_MXS_VIIM=y
 
 #
@@ -985,57 +799,13 @@ CONFIG_UNIX98_PTYS=y
 CONFIG_LEGACY_PTYS=y
 CONFIG_LEGACY_PTY_COUNT=256
 # CONFIG_FSL_OTP is not set
-CONFIG_IPMI_HANDLER=m
-# CONFIG_IPMI_PANIC_EVENT is not set
-CONFIG_IPMI_DEVICE_INTERFACE=m
-CONFIG_IPMI_SI=m
-CONFIG_IPMI_WATCHDOG=m
-CONFIG_IPMI_POWEROFF=m
-CONFIG_HW_RANDOM=m
-# CONFIG_HW_RANDOM_TIMERIOMEM is not set
+# CONFIG_IPMI_HANDLER is not set
+# CONFIG_HW_RANDOM is not set
 # CONFIG_R3964 is not set
-CONFIG_RAW_DRIVER=y
-CONFIG_MAX_RAW_DEVS=8192
+# CONFIG_RAW_DRIVER is not set
 # CONFIG_TCG_TPM is not set
 # CONFIG_RAMOOPS is not set
-CONFIG_I2C=y
-CONFIG_I2C_BOARDINFO=y
-CONFIG_I2C_COMPAT=y
-# CONFIG_I2C_CHARDEV is not set
-CONFIG_I2C_HELPER_AUTO=y
-
-#
-# I2C Hardware Bus support
-#
-
-#
-# I2C system bus drivers (mostly embedded / system-on-chip)
-#
-# CONFIG_I2C_DESIGNWARE is not set
-# CONFIG_I2C_GPIO is not set
-CONFIG_I2C_MXS=y
-CONFIG_I2C_MXS_SELECT0=y
-# CONFIG_I2C_MXS_SELECT0_PIOQUEUE_MODE is not set
-# CONFIG_I2C_MXS_SELECT1 is not set
-# CONFIG_I2C_OCORES is not set
-# CONFIG_I2C_PCA_PLATFORM is not set
-# CONFIG_I2C_SIMTEC is not set
-# CONFIG_I2C_XILINX is not set
-
-#
-# External I2C/SMBus adapter drivers
-#
-# CONFIG_I2C_PARPORT_LIGHT is not set
-# CONFIG_I2C_TAOS_EVM is not set
-# CONFIG_I2C_TINY_USB is not set
-
-#
-# Other I2C/SMBus bus drivers
-#
-# CONFIG_I2C_STUB is not set
-# CONFIG_I2C_DEBUG_CORE is not set
-# CONFIG_I2C_DEBUG_ALGO is not set
-# CONFIG_I2C_DEBUG_BUS is not set
+# CONFIG_I2C is not set
 CONFIG_SPI=y
 CONFIG_SPI_MASTER=y
 
@@ -1070,11 +840,6 @@ CONFIG_GPIOLIB=y
 #
 # I2C GPIO expanders:
 #
-# CONFIG_GPIO_MAX7300 is not set
-# CONFIG_GPIO_MAX732X is not set
-# CONFIG_GPIO_PCA953X is not set
-# CONFIG_GPIO_PCF857X is not set
-# CONFIG_GPIO_ADP5588 is not set
 
 #
 # PCI GPIO expanders:
@@ -1100,9 +865,6 @@ CONFIG_POWER_SUPPLY=y
 # CONFIG_PDA_POWER is not set
 # CONFIG_TEST_POWER is not set
 # CONFIG_BATTERY_DS2760 is not set
-# CONFIG_BATTERY_DS2782 is not set
-# CONFIG_BATTERY_BQ27x00 is not set
-# CONFIG_BATTERY_MAX17040 is not set
 CONFIG_BATTERY_MXS=y
 # CONFIG_MXS_VBUS_CURRENT_DRAW is not set
 # CONFIG_BATTERY_DS2438 is not set
@@ -1117,11 +879,6 @@ CONFIG_WATCHDOG=y
 # CONFIG_SOFT_WATCHDOG is not set
 CONFIG_MXS_WATCHDOG=y
 # CONFIG_MAX63XX_WATCHDOG is not set
-
-#
-# USB-based Watchdog Cards
-#
-# CONFIG_USBPCWATCHDOG is not set
 CONFIG_SSB_POSSIBLE=y
 
 #
@@ -1131,36 +888,7 @@ CONFIG_SSB=m
 CONFIG_SSB_SDIOHOST_POSSIBLE=y
 # CONFIG_SSB_SDIOHOST is not set
 # CONFIG_SSB_DEBUG is not set
-CONFIG_MFD_SUPPORT=y
-# CONFIG_MFD_CORE is not set
-# CONFIG_MFD_88PM860X is not set
-# CONFIG_MFD_SM501 is not set
-# CONFIG_MFD_ASIC3 is not set
-# CONFIG_HTC_EGPIO is not set
-# CONFIG_HTC_PASIC3 is not set
-# CONFIG_HTC_I2CPLD is not set
-# CONFIG_TPS65010 is not set
-# CONFIG_TPS6507X is not set
-# CONFIG_TWL4030_CORE is not set
-# CONFIG_MFD_TC35892 is not set
-# CONFIG_MFD_TMIO is not set
-# CONFIG_MFD_T7L66XB is not set
-# CONFIG_MFD_TC6387XB is not set
-# CONFIG_MFD_TC6393XB is not set
-# CONFIG_PMIC_DA903X is not set
-# CONFIG_PMIC_ADP5520 is not set
-# CONFIG_MFD_MAX8925 is not set
-# CONFIG_MFD_WM8400 is not set
-# CONFIG_MFD_WM831X is not set
-# CONFIG_MFD_WM8350_I2C is not set
-# CONFIG_MFD_WM8994 is not set
-# CONFIG_MFD_LTC3589_I2C is not set
-# CONFIG_MFD_PCF50633 is not set
-# CONFIG_MFD_MC13783 is not set
-# CONFIG_ABX500_CORE is not set
-# CONFIG_EZX_PCAP is not set
-# CONFIG_AB8500_CORE is not set
-# CONFIG_MFD_MAX17135 is not set
+# CONFIG_MFD_SUPPORT is not set
 CONFIG_REGULATOR=y
 # CONFIG_REGULATOR_DEBUG is not set
 # CONFIG_REGULATOR_DUMMY is not set
@@ -1168,182 +896,17 @@ CONFIG_REGULATOR=y
 # CONFIG_REGULATOR_VIRTUAL_CONSUMER is not set
 # CONFIG_REGULATOR_USERSPACE_CONSUMER is not set
 # CONFIG_REGULATOR_BQ24022 is not set
-# CONFIG_REGULATOR_MAX1586 is not set
-# CONFIG_REGULATOR_MAX8649 is not set
-# CONFIG_REGULATOR_MAX8660 is not set
-# CONFIG_REGULATOR_LP3971 is not set
-# CONFIG_REGULATOR_TPS65023 is not set
-# CONFIG_REGULATOR_TPS6507X is not set
 CONFIG_REGULATOR_MXS=y
-CONFIG_MEDIA_SUPPORT=y
-
-#
-# Multimedia core support
-#
-CONFIG_VIDEO_DEV=y
-CONFIG_VIDEO_V4L2_COMMON=y
-# CONFIG_VIDEO_ALLOW_V4L1 is not set
-CONFIG_VIDEO_V4L1_COMPAT=y
-# CONFIG_DVB_CORE is not set
-CONFIG_VIDEO_MEDIA=y
-
-#
-# Multimedia drivers
-#
-CONFIG_IR_CORE=y
-CONFIG_VIDEO_IR=y
-CONFIG_RC_MAP=y
-CONFIG_IR_NEC_DECODER=y
-CONFIG_IR_RC5_DECODER=y
-CONFIG_IR_RC6_DECODER=y
-CONFIG_IR_JVC_DECODER=y
-CONFIG_IR_SONY_DECODER=y
-# CONFIG_IR_IMON is not set
-# CONFIG_MEDIA_ATTACH is not set
-CONFIG_MEDIA_TUNER=y
-# CONFIG_MEDIA_TUNER_CUSTOMISE is not set
-CONFIG_MEDIA_TUNER_SIMPLE=y
-CONFIG_MEDIA_TUNER_TDA8290=y
-CONFIG_MEDIA_TUNER_TDA9887=y
-CONFIG_MEDIA_TUNER_TEA5761=y
-CONFIG_MEDIA_TUNER_TEA5767=y
-CONFIG_MEDIA_TUNER_MT20XX=y
-CONFIG_MEDIA_TUNER_XC2028=y
-CONFIG_MEDIA_TUNER_XC5000=y
-CONFIG_MEDIA_TUNER_MC44S803=y
-CONFIG_VIDEO_V4L2=y
-CONFIG_VIDEOBUF_GEN=y
-CONFIG_VIDEOBUF_DMA_CONTIG=y
-CONFIG_VIDEO_CAPTURE_DRIVERS=y
-# CONFIG_VIDEO_ADV_DEBUG is not set
-# CONFIG_VIDEO_FIXED_MINOR_RANGES is not set
-# CONFIG_VIDEO_HELPER_CHIPS_AUTO is not set
-CONFIG_VIDEO_IR_I2C=y
-
-#
-# Encoders/decoders and other helper chips
-#
-
-#
-# Audio decoders
-#
-# CONFIG_VIDEO_TVAUDIO is not set
-# CONFIG_VIDEO_TDA7432 is not set
-# CONFIG_VIDEO_TDA9840 is not set
-# CONFIG_VIDEO_TDA9875 is not set
-# CONFIG_VIDEO_TEA6415C is not set
-# CONFIG_VIDEO_TEA6420 is not set
-# CONFIG_VIDEO_MSP3400 is not set
-# CONFIG_VIDEO_CS5345 is not set
-# CONFIG_VIDEO_CS53L32A is not set
-# CONFIG_VIDEO_M52790 is not set
-# CONFIG_VIDEO_TLV320AIC23B is not set
-# CONFIG_VIDEO_WM8775 is not set
-# CONFIG_VIDEO_WM8739 is not set
-# CONFIG_VIDEO_VP27SMPX is not set
-
-#
-# RDS decoders
-#
-# CONFIG_VIDEO_SAA6588 is not set
-
-#
-# Video decoders
-#
-# CONFIG_VIDEO_ADV7180 is not set
-# CONFIG_VIDEO_BT819 is not set
-# CONFIG_VIDEO_BT856 is not set
-# CONFIG_VIDEO_BT866 is not set
-# CONFIG_VIDEO_KS0127 is not set
-# CONFIG_VIDEO_OV7670 is not set
-# CONFIG_VIDEO_MT9V011 is not set
-# CONFIG_VIDEO_TCM825X is not set
-# CONFIG_VIDEO_SAA7110 is not set
-# CONFIG_VIDEO_SAA711X is not set
-# CONFIG_VIDEO_SAA717X is not set
-# CONFIG_VIDEO_SAA7191 is not set
-# CONFIG_VIDEO_TVP514X is not set
-# CONFIG_VIDEO_TVP5150 is not set
-# CONFIG_VIDEO_TVP7002 is not set
-# CONFIG_VIDEO_VPX3220 is not set
-
-#
-# Video and audio decoders
-#
-# CONFIG_VIDEO_CX25840 is not set
-
-#
-# MPEG video encoders
-#
-# CONFIG_VIDEO_CX2341X is not set
-
-#
-# Video encoders
-#
-# CONFIG_VIDEO_SAA7127 is not set
-# CONFIG_VIDEO_SAA7185 is not set
-# CONFIG_VIDEO_ADV7170 is not set
-# CONFIG_VIDEO_ADV7175 is not set
-# CONFIG_VIDEO_THS7303 is not set
-# CONFIG_VIDEO_ADV7343 is not set
-# CONFIG_VIDEO_AK881X is not set
-
-#
-# Video improvement chips
-#
-# CONFIG_VIDEO_UPD64031A is not set
-# CONFIG_VIDEO_UPD64083 is not set
-CONFIG_VIDEO_MXS_PXP=y
-# CONFIG_VIDEO_SAA5246A is not set
-# CONFIG_VIDEO_SAA5249 is not set
-# CONFIG_SOC_CAMERA is not set
-# CONFIG_V4L_USB_DRIVERS is not set
-# CONFIG_V4L_MEM2MEM_DRIVERS is not set
-# CONFIG_RADIO_ADAPTERS is not set
-# CONFIG_DAB is not set
+# CONFIG_MEDIA_SUPPORT is not set
 
 #
 # Graphics support
 #
 # CONFIG_VGASTATE is not set
 # CONFIG_VIDEO_OUTPUT_CONTROL is not set
-CONFIG_FB=y
-# CONFIG_FIRMWARE_EDID is not set
-# CONFIG_FB_DDC is not set
-# CONFIG_FB_BOOT_VESA_SUPPORT is not set
-CONFIG_FB_CFB_FILLRECT=y
-CONFIG_FB_CFB_COPYAREA=y
-CONFIG_FB_CFB_IMAGEBLIT=y
-# CONFIG_FB_CFB_REV_PIXELS_IN_BYTE is not set
-# CONFIG_FB_SYS_FILLRECT is not set
-# CONFIG_FB_SYS_COPYAREA is not set
-# CONFIG_FB_SYS_IMAGEBLIT is not set
-# CONFIG_FB_FOREIGN_ENDIAN is not set
-# CONFIG_FB_SYS_FOPS is not set
-# CONFIG_FB_SVGALIB is not set
-# CONFIG_FB_MACMODES is not set
-# CONFIG_FB_BACKLIGHT is not set
-# CONFIG_FB_MODE_HELPERS is not set
-# CONFIG_FB_TILEBLITTING is not set
-
-#
-# Frame buffer hardware drivers
-#
-CONFIG_FB_MXS=y
-CONFIG_FB_MXS_LCD_43WVF1G=y
-# CONFIG_FB_MXS_LCD_LMS430 is not set
+# CONFIG_FB is not set
 # CONFIG_FB_MXS_TVENC is not set
-# CONFIG_FB_S1D13XXX is not set
-# CONFIG_FB_VIRTUAL is not set
-# CONFIG_FB_METRONOME is not set
-# CONFIG_FB_MB862XX is not set
-# CONFIG_FB_BROADSHEET is not set
-CONFIG_BACKLIGHT_LCD_SUPPORT=y
-# CONFIG_LCD_CLASS_DEVICE is not set
-CONFIG_BACKLIGHT_CLASS_DEVICE=y
-# CONFIG_BACKLIGHT_GENERIC is not set
-# CONFIG_BACKLIGHT_ADP8860 is not set
-CONFIG_BACKLIGHT_MXS=y
+# CONFIG_BACKLIGHT_LCD_SUPPORT is not set
 
 #
 # Display device support
@@ -1355,281 +918,13 @@ CONFIG_BACKLIGHT_MXS=y
 #
 # CONFIG_VGA_CONSOLE is not set
 CONFIG_DUMMY_CONSOLE=y
-CONFIG_FRAMEBUFFER_CONSOLE=y
-# CONFIG_FRAMEBUFFER_CONSOLE_DETECT_PRIMARY is not set
-# CONFIG_FRAMEBUFFER_CONSOLE_ROTATION is not set
-# CONFIG_FONTS is not set
-CONFIG_FONT_8x8=y
-CONFIG_FONT_8x16=y
-CONFIG_LOGO=y
-# CONFIG_LOGO_LINUX_MONO is not set
-# CONFIG_LOGO_LINUX_VGA16 is not set
-CONFIG_LOGO_LINUX_CLUT224=y
-CONFIG_SOUND=y
-# CONFIG_SOUND_OSS_CORE is not set
-CONFIG_SND=y
-CONFIG_SND_TIMER=y
-CONFIG_SND_PCM=y
-CONFIG_SND_JACK=y
-# CONFIG_SND_SEQUENCER is not set
-# CONFIG_SND_MIXER_OSS is not set
-# CONFIG_SND_PCM_OSS is not set
-# CONFIG_SND_HRTIMER is not set
-# CONFIG_SND_DYNAMIC_MINORS is not set
-CONFIG_SND_SUPPORT_OLD_API=y
-CONFIG_SND_VERBOSE_PROCFS=y
-# CONFIG_SND_VERBOSE_PRINTK is not set
-# CONFIG_SND_DEBUG is not set
-# CONFIG_SND_RAWMIDI_SEQ is not set
-# CONFIG_SND_OPL3_LIB_SEQ is not set
-# CONFIG_SND_OPL4_LIB_SEQ is not set
-# CONFIG_SND_SBAWE_SEQ is not set
-# CONFIG_SND_EMU10K1_SEQ is not set
-# CONFIG_SND_DRIVERS is not set
-# CONFIG_SND_ARM is not set
-CONFIG_SND_SPI=y
-# CONFIG_SND_USB is not set
-CONFIG_SND_SOC=y
-CONFIG_SND_MXS_SOC=y
-CONFIG_SND_MXS_SOC_SPDIF_DAI=y
-CONFIG_SND_MXS_SOC_EVK_DEVB=y
-CONFIG_SND_MXS_SOC_DAI=y
-CONFIG_SND_MXS_SOC_EVK_DEVB_SPDIF=y
-CONFIG_SND_SOC_I2C_AND_SPI=y
-# CONFIG_SND_SOC_ALL_CODECS is not set
-CONFIG_SND_SOC_SGTL5000=y
-CONFIG_SND_SOC_MXS_SPDIF=y
-# CONFIG_SOUND_PRIME is not set
-CONFIG_HID_SUPPORT=y
-CONFIG_HID=y
-# CONFIG_HIDRAW is not set
-
-#
-# USB Input Devices
-#
-CONFIG_USB_HID=y
-# CONFIG_HID_PID is not set
-CONFIG_USB_HIDDEV=y
-
-#
-# Special HID drivers
-#
-# CONFIG_HID_3M_PCT is not set
-CONFIG_HID_A4TECH=y
-CONFIG_HID_APPLE=y
-CONFIG_HID_BELKIN=y
-# CONFIG_HID_CANDO is not set
-CONFIG_HID_CHERRY=y
-CONFIG_HID_CHICONY=y
-# CONFIG_HID_PRODIKEYS is not set
-CONFIG_HID_CYPRESS=y
-CONFIG_HID_DRAGONRISE=y
-# CONFIG_DRAGONRISE_FF is not set
-# CONFIG_HID_EGALAX is not set
-CONFIG_HID_EZKEY=y
-CONFIG_HID_KYE=y
-CONFIG_HID_GYRATION=y
-# CONFIG_HID_TWINHAN is not set
-CONFIG_HID_KENSINGTON=y
-CONFIG_HID_LOGITECH=y
-# CONFIG_LOGITECH_FF is not set
-# CONFIG_LOGIRUMBLEPAD2_FF is not set
-# CONFIG_LOGIG940_FF is not set
-CONFIG_HID_MICROSOFT=y
-# CONFIG_HID_MOSART is not set
-CONFIG_HID_MONTEREY=y
-CONFIG_HID_NTRIG=y
-# CONFIG_HID_ORTEK is not set
-CONFIG_HID_PANTHERLORD=y
-# CONFIG_PANTHERLORD_FF is not set
-CONFIG_HID_PETALYNX=y
-# CONFIG_HID_PICOLCD is not set
-# CONFIG_HID_QUANTA is not set
-# CONFIG_HID_ROCCAT is not set
-# CONFIG_HID_ROCCAT_KONE is not set
-CONFIG_HID_SAMSUNG=y
-CONFIG_HID_SONY=y
-# CONFIG_HID_STANTUM is not set
-CONFIG_HID_SUNPLUS=y
-CONFIG_HID_GREENASIA=y
-# CONFIG_GREENASIA_FF is not set
-CONFIG_HID_SMARTJOYPLUS=y
-# CONFIG_SMARTJOYPLUS_FF is not set
-CONFIG_HID_TOPSEED=y
-CONFIG_HID_THRUSTMASTER=y
-# CONFIG_THRUSTMASTER_FF is not set
-CONFIG_HID_ZEROPLUS=y
-# CONFIG_ZEROPLUS_FF is not set
-# CONFIG_HID_ZYDACRON is not set
-CONFIG_USB_SUPPORT=y
-CONFIG_USB_ARCH_HAS_HCD=y
-# CONFIG_USB_ARCH_HAS_OHCI is not set
+# CONFIG_SOUND is not set
+# CONFIG_HID_SUPPORT is not set
+# CONFIG_USB_SUPPORT is not set
 CONFIG_USB_ARCH_HAS_EHCI=y
-CONFIG_USB=y
-# CONFIG_USB_DEBUG is not set
-# CONFIG_USB_ANNOUNCE_NEW_DEVICES is not set
-
-#
-# Miscellaneous USB options
-#
-# CONFIG_USB_DEVICEFS is not set
-# CONFIG_USB_DEVICE_CLASS is not set
-# CONFIG_USB_DYNAMIC_MINORS is not set
-CONFIG_USB_SUSPEND=y
-CONFIG_USB_OTG=y
-# CONFIG_USB_OTG_WHITELIST is not set
-# CONFIG_USB_OTG_BLACKLIST_HUB is not set
-# CONFIG_USB_MON is not set
-# CONFIG_USB_WUSB is not set
-# CONFIG_USB_WUSB_CBAF is not set
-
-#
-# USB Host Controller Drivers
-#
-# CONFIG_USB_C67X00_HCD is not set
-CONFIG_USB_EHCI_HCD=y
-CONFIG_USB_EHCI_ARC=y
-CONFIG_USB_EHCI_ARC_H1=y
-CONFIG_USB_EHCI_ARC_OTG=y
-# CONFIG_USB_STATIC_IRAM is not set
-# CONFIG_USB_EHCI_FSL_MC13783 is not set
-# CONFIG_USB_EHCI_FSL_1301 is not set
-# CONFIG_USB_EHCI_FSL_1504 is not set
-CONFIG_USB_EHCI_FSL_UTMI=y
-CONFIG_USB_EHCI_ROOT_HUB_TT=y
-# CONFIG_USB_EHCI_TT_NEWSCHED is not set
-# CONFIG_USB_OXU210HP_HCD is not set
-# CONFIG_USB_ISP116X_HCD is not set
-# CONFIG_USB_ISP1760_HCD is not set
-# CONFIG_USB_ISP1362_HCD is not set
-# CONFIG_USB_SL811_HCD is not set
-# CONFIG_USB_R8A66597_HCD is not set
-# CONFIG_USB_HWA_HCD is not set
-# CONFIG_USB_MUSB_HDRC is not set
-# CONFIG_USB_GADGET_MUSB_HDRC is not set
-
-#
-# USB Device Class drivers
-#
-# CONFIG_USB_ACM is not set
-# CONFIG_USB_PRINTER is not set
-# CONFIG_USB_WDM is not set
-# CONFIG_USB_TMC is not set
-
-#
-# NOTE: USB_STORAGE depends on SCSI but BLK_DEV_SD may
-#
-
-#
-# also be needed; see USB_STORAGE Help for more info
-#
-CONFIG_USB_STORAGE=y
-# CONFIG_USB_STORAGE_DEBUG is not set
-# CONFIG_USB_STORAGE_DATAFAB is not set
-# CONFIG_USB_STORAGE_FREECOM is not set
-# CONFIG_USB_STORAGE_ISD200 is not set
-# CONFIG_USB_STORAGE_USBAT is not set
-# CONFIG_USB_STORAGE_SDDR09 is not set
-# CONFIG_USB_STORAGE_SDDR55 is not set
-# CONFIG_USB_STORAGE_JUMPSHOT is not set
-# CONFIG_USB_STORAGE_ALAUDA is not set
-# CONFIG_USB_STORAGE_ONETOUCH is not set
-# CONFIG_USB_STORAGE_KARMA is not set
-# CONFIG_USB_STORAGE_CYPRESS_ATACB is not set
-# CONFIG_USB_LIBUSUAL is not set
-
-#
-# USB Imaging devices
-#
-# CONFIG_USB_MDC800 is not set
-# CONFIG_USB_MICROTEK is not set
-
-#
-# USB port drivers
-#
-# CONFIG_USB_SERIAL is not set
-
-#
-# USB Miscellaneous drivers
-#
-# CONFIG_USB_EMI62 is not set
-# CONFIG_USB_EMI26 is not set
-# CONFIG_USB_ADUTUX is not set
-# CONFIG_USB_SEVSEG is not set
-# CONFIG_USB_RIO500 is not set
-# CONFIG_USB_LEGOTOWER is not set
-# CONFIG_USB_LCD is not set
-# CONFIG_USB_LED is not set
-# CONFIG_USB_CYPRESS_CY7C63 is not set
-# CONFIG_USB_CYTHERM is not set
-# CONFIG_USB_IDMOUSE is not set
-# CONFIG_USB_FTDI_ELAN is not set
-# CONFIG_USB_APPLEDISPLAY is not set
-# CONFIG_USB_SISUSBVGA is not set
-# CONFIG_USB_LD is not set
-# CONFIG_USB_TRANCEVIBRATOR is not set
-# CONFIG_USB_IOWARRIOR is not set
-# CONFIG_USB_TEST is not set
-# CONFIG_USB_ISIGHTFW is not set
-CONFIG_USB_GADGET=y
-# CONFIG_USB_GADGET_DEBUG_FILES is not set
-# CONFIG_USB_GADGET_DEBUG_FS is not set
-CONFIG_USB_GADGET_VBUS_DRAW=2
-CONFIG_USB_GADGET_SELECTED=y
-# CONFIG_USB_GADGET_AT91 is not set
-# CONFIG_USB_GADGET_ATMEL_USBA is not set
-# CONFIG_USB_GADGET_FSL_USB2 is not set
-# CONFIG_USB_GADGET_LH7A40X is not set
-# CONFIG_USB_GADGET_OMAP is not set
-# CONFIG_USB_GADGET_PXA25X is not set
-# CONFIG_USB_GADGET_R8A66597 is not set
-# CONFIG_USB_GADGET_PXA27X is not set
-# CONFIG_USB_GADGET_S3C_HSOTG is not set
-# CONFIG_USB_GADGET_IMX is not set
-# CONFIG_USB_GADGET_S3C2410 is not set
-# CONFIG_USB_GADGET_M66592 is not set
-# CONFIG_USB_GADGET_AMD5536UDC is not set
-# CONFIG_USB_GADGET_FSL_QE is not set
-# CONFIG_USB_GADGET_CI13XXX is not set
-# CONFIG_USB_GADGET_NET2280 is not set
-# CONFIG_USB_GADGET_GOKU is not set
-CONFIG_USB_GADGET_ARC=y
-CONFIG_USB_ARC=y
-CONFIG_WORKAROUND_ARCUSB_REG_RW=y
-# CONFIG_USB_GADGET_LANGWELL is not set
-# CONFIG_USB_GADGET_DUMMY_HCD is not set
-CONFIG_USB_GADGET_DUALSPEED=y
-# CONFIG_USB_ZERO is not set
-# CONFIG_USB_AUDIO is not set
-CONFIG_USB_ETH=m
-CONFIG_USB_ETH_RNDIS=y
-# CONFIG_USB_ETH_EEM is not set
-# CONFIG_USB_GADGETFS is not set
-# CONFIG_USB_FUNCTIONFS is not set
-CONFIG_USB_FILE_STORAGE=m
-# CONFIG_FSL_UTP is not set
-# CONFIG_USB_FILE_STORAGE_TEST is not set
-# CONFIG_USB_MASS_STORAGE is not set
-CONFIG_USB_G_SERIAL=m
-# CONFIG_USB_MIDI_GADGET is not set
-# CONFIG_USB_G_PRINTER is not set
-# CONFIG_USB_CDC_COMPOSITE is not set
-# CONFIG_USB_G_NOKIA is not set
-# CONFIG_USB_G_MULTI is not set
-# CONFIG_USB_G_HID is not set
-# CONFIG_USB_G_WEBCAM is not set
-
-#
-# OTG and related infrastructure
-#
-CONFIG_USB_OTG_UTILS=y
-# CONFIG_USB_GPIO_VBUS is not set
-# CONFIG_USB_ULPI is not set
-# CONFIG_NOP_USB_XCEIV is not set
-CONFIG_MXC_OTG=y
 CONFIG_MMC=y
 # CONFIG_MMC_DEBUG is not set
-CONFIG_MMC_UNSAFE_RESUME=y
+# CONFIG_MMC_UNSAFE_RESUME is not set
 
 #
 # MMC/SD/SDIO Card Drivers
@@ -1654,13 +949,9 @@ CONFIG_LEDS_CLASS=y
 # LED drivers
 #
 CONFIG_LEDS_MXS=y
-# CONFIG_LEDS_PCA9532 is not set
 # CONFIG_LEDS_GPIO is not set
-# CONFIG_LEDS_LP3944 is not set
-# CONFIG_LEDS_PCA955X is not set
 # CONFIG_LEDS_DAC124S085 is not set
 # CONFIG_LEDS_REGULATOR is not set
-# CONFIG_LEDS_BD2802 is not set
 # CONFIG_LEDS_LT3593 is not set
 # CONFIG_LEDS_TRIGGERS is not set
 # CONFIG_ACCESSIBILITY is not set
@@ -1680,25 +971,6 @@ CONFIG_RTC_INTF_DEV=y
 # CONFIG_RTC_DRV_TEST is not set
 
 #
-# I2C RTC drivers
-#
-# CONFIG_RTC_DRV_DS1307 is not set
-# CONFIG_RTC_DRV_DS1374 is not set
-# CONFIG_RTC_DRV_DS1672 is not set
-# CONFIG_RTC_DRV_MAX6900 is not set
-# CONFIG_RTC_DRV_RS5C372 is not set
-# CONFIG_RTC_DRV_ISL1208 is not set
-# CONFIG_RTC_DRV_X1205 is not set
-# CONFIG_RTC_DRV_PCF8563 is not set
-# CONFIG_RTC_DRV_PCF8583 is not set
-# CONFIG_RTC_DRV_M41T80 is not set
-# CONFIG_RTC_DRV_BQ32K is not set
-# CONFIG_RTC_DRV_S35390A is not set
-# CONFIG_RTC_DRV_FM3130 is not set
-# CONFIG_RTC_DRV_RX8581 is not set
-# CONFIG_RTC_DRV_RX8025 is not set
-
-#
 # SPI RTC drivers
 #
 # CONFIG_RTC_DRV_M41T94 is not set
@@ -1751,14 +1023,8 @@ CONFIG_EXT2_FS_XATTR=y
 CONFIG_EXT2_FS_POSIX_ACL=y
 CONFIG_EXT2_FS_SECURITY=y
 # CONFIG_EXT2_FS_XIP is not set
-CONFIG_EXT3_FS=y
-CONFIG_EXT3_DEFAULTS_TO_ORDERED=y
-CONFIG_EXT3_FS_XATTR=y
-CONFIG_EXT3_FS_POSIX_ACL=y
-CONFIG_EXT3_FS_SECURITY=y
+# CONFIG_EXT3_FS is not set
 # CONFIG_EXT4_FS is not set
-CONFIG_JBD=y
-# CONFIG_JBD_DEBUG is not set
 CONFIG_FS_MBCACHE=y
 # CONFIG_REISERFS_FS is not set
 # CONFIG_JFS_FS is not set
@@ -1782,7 +1048,7 @@ CONFIG_QUOTA_TREE=y
 CONFIG_QFMT_V2=y
 CONFIG_QUOTACTL=y
 # CONFIG_AUTOFS_FS is not set
-CONFIG_AUTOFS4_FS=m
+# CONFIG_AUTOFS4_FS is not set
 # CONFIG_FUSE_FS is not set
 
 #
@@ -1793,20 +1059,14 @@ CONFIG_AUTOFS4_FS=m
 #
 # CD-ROM/DVD Filesystems
 #
-CONFIG_ISO9660_FS=y
-CONFIG_JOLIET=y
-CONFIG_ZISOFS=y
-CONFIG_UDF_FS=m
-CONFIG_UDF_NLS=y
+# CONFIG_ISO9660_FS is not set
+# CONFIG_UDF_FS is not set
 
 #
 # DOS/FAT/NT Filesystems
 #
-CONFIG_FAT_FS=y
-CONFIG_MSDOS_FS=y
-CONFIG_VFAT_FS=y
-CONFIG_FAT_DEFAULT_CODEPAGE=437
-CONFIG_FAT_DEFAULT_IOCHARSET="ascii"
+# CONFIG_MSDOS_FS is not set
+# CONFIG_VFAT_FS is not set
 # CONFIG_NTFS_FS is not set
 
 #
@@ -1824,8 +1084,8 @@ CONFIG_MISC_FILESYSTEMS=y
 # CONFIG_ADFS_FS is not set
 # CONFIG_AFFS_FS is not set
 # CONFIG_ECRYPT_FS is not set
-CONFIG_HFS_FS=m
-CONFIG_HFSPLUS_FS=m
+# CONFIG_HFS_FS is not set
+# CONFIG_HFSPLUS_FS is not set
 # CONFIG_BEFS_FS is not set
 # CONFIG_BFS_FS is not set
 # CONFIG_EFS_FS is not set
@@ -1840,16 +1100,11 @@ CONFIG_JFFS2_ZLIB=y
 # CONFIG_JFFS2_LZO is not set
 CONFIG_JFFS2_RTIME=y
 # CONFIG_JFFS2_RUBIN is not set
-CONFIG_UBIFS_FS=y
-# CONFIG_UBIFS_FS_XATTR is not set
-# CONFIG_UBIFS_FS_ADVANCED_COMPR is not set
-CONFIG_UBIFS_FS_LZO=y
-CONFIG_UBIFS_FS_ZLIB=y
-# CONFIG_UBIFS_FS_DEBUG is not set
+# CONFIG_UBIFS_FS is not set
 # CONFIG_LOGFS is not set
-CONFIG_CRAMFS=m
+# CONFIG_CRAMFS is not set
 # CONFIG_SQUASHFS is not set
-CONFIG_VXFS_FS=m
+# CONFIG_VXFS_FS is not set
 # CONFIG_MINIX_FS is not set
 # CONFIG_OMFS_FS is not set
 # CONFIG_HPFS_FS is not set
@@ -1880,24 +1135,8 @@ CONFIG_SUNRPC=y
 #
 # Partition Types
 #
-CONFIG_PARTITION_ADVANCED=y
-# CONFIG_ACORN_PARTITION is not set
-CONFIG_OSF_PARTITION=y
-# CONFIG_AMIGA_PARTITION is not set
-# CONFIG_ATARI_PARTITION is not set
-CONFIG_MAC_PARTITION=y
+# CONFIG_PARTITION_ADVANCED is not set
 CONFIG_MSDOS_PARTITION=y
-CONFIG_BSD_DISKLABEL=y
-CONFIG_MINIX_SUBPARTITION=y
-CONFIG_SOLARIS_X86_PARTITION=y
-CONFIG_UNIXWARE_DISKLABEL=y
-# CONFIG_LDM_PARTITION is not set
-CONFIG_SGI_PARTITION=y
-# CONFIG_ULTRIX_PARTITION is not set
-CONFIG_SUN_PARTITION=y
-# CONFIG_KARMA_PARTITION is not set
-CONFIG_EFI_PARTITION=y
-# CONFIG_SYSV68_PARTITION is not set
 CONFIG_NLS=y
 CONFIG_NLS_DEFAULT="utf8"
 CONFIG_NLS_CODEPAGE_437=y
@@ -1985,22 +1224,14 @@ CONFIG_SECURITY=y
 CONFIG_SECURITY_NETWORK=y
 # CONFIG_SECURITY_NETWORK_XFRM is not set
 # CONFIG_SECURITY_PATH is not set
-CONFIG_LSM_MMAP_MIN_ADDR=65536
-CONFIG_SECURITY_SELINUX=y
-CONFIG_SECURITY_SELINUX_BOOTPARAM=y
-CONFIG_SECURITY_SELINUX_BOOTPARAM_VALUE=1
-CONFIG_SECURITY_SELINUX_DISABLE=y
-CONFIG_SECURITY_SELINUX_DEVELOP=y
-CONFIG_SECURITY_SELINUX_AVC_STATS=y
-CONFIG_SECURITY_SELINUX_CHECKREQPROT_VALUE=1
-# CONFIG_SECURITY_SELINUX_POLICYDB_VERSION_MAX is not set
+# CONFIG_SECURITY_SELINUX is not set
 # CONFIG_SECURITY_TOMOYO is not set
 # CONFIG_IMA is not set
-CONFIG_DEFAULT_SECURITY_SELINUX=y
+# CONFIG_DEFAULT_SECURITY_SELINUX is not set
 # CONFIG_DEFAULT_SECURITY_SMACK is not set
 # CONFIG_DEFAULT_SECURITY_TOMOYO is not set
-# CONFIG_DEFAULT_SECURITY_DAC is not set
-CONFIG_DEFAULT_SECURITY="selinux"
+CONFIG_DEFAULT_SECURITY_DAC=y
+CONFIG_DEFAULT_SECURITY=""
 CONFIG_CRYPTO=y
 
 #
@@ -2008,23 +1239,16 @@ CONFIG_CRYPTO=y
 #
 CONFIG_CRYPTO_ALGAPI=y
 CONFIG_CRYPTO_ALGAPI2=y
-CONFIG_CRYPTO_AEAD2=y
-CONFIG_CRYPTO_BLKCIPHER=y
-CONFIG_CRYPTO_BLKCIPHER2=y
-CONFIG_CRYPTO_HASH=y
-CONFIG_CRYPTO_HASH2=y
-CONFIG_CRYPTO_RNG2=y
-CONFIG_CRYPTO_PCOMP=y
-CONFIG_CRYPTO_MANAGER=y
-CONFIG_CRYPTO_MANAGER2=y
-CONFIG_CRYPTO_MANAGER_TESTS=y
+CONFIG_CRYPTO_HASH=m
+CONFIG_CRYPTO_HASH2=m
+# CONFIG_CRYPTO_MANAGER is not set
+# CONFIG_CRYPTO_MANAGER2 is not set
 # CONFIG_CRYPTO_GF128MUL is not set
 # CONFIG_CRYPTO_NULL is not set
-CONFIG_CRYPTO_WORKQUEUE=y
 # CONFIG_CRYPTO_CRYPTD is not set
 # CONFIG_CRYPTO_AUTHENC is not set
-CONFIG_CRYPTO_TEST=m
-CONFIG_CRYPTO_CRYPTODEV=y
+# CONFIG_CRYPTO_TEST is not set
+# CONFIG_CRYPTO_CRYPTODEV is not set
 
 #
 # Authenticated Encryption with Associated Data
@@ -2036,10 +1260,10 @@ CONFIG_CRYPTO_CRYPTODEV=y
 #
 # Block modes
 #
-CONFIG_CRYPTO_CBC=y
+# CONFIG_CRYPTO_CBC is not set
 # CONFIG_CRYPTO_CTR is not set
 # CONFIG_CRYPTO_CTS is not set
-CONFIG_CRYPTO_ECB=y
+# CONFIG_CRYPTO_ECB is not set
 # CONFIG_CRYPTO_LRW is not set
 # CONFIG_CRYPTO_PCBC is not set
 # CONFIG_CRYPTO_XTS is not set
@@ -2047,7 +1271,7 @@ CONFIG_CRYPTO_ECB=y
 #
 # Hash modes
 #
-CONFIG_CRYPTO_HMAC=y
+# CONFIG_CRYPTO_HMAC is not set
 # CONFIG_CRYPTO_XCBC is not set
 # CONFIG_CRYPTO_VMAC is not set
 
@@ -2057,14 +1281,14 @@ CONFIG_CRYPTO_HMAC=y
 CONFIG_CRYPTO_CRC32C=m
 # CONFIG_CRYPTO_GHASH is not set
 # CONFIG_CRYPTO_MD4 is not set
-CONFIG_CRYPTO_MD5=y
+# CONFIG_CRYPTO_MD5 is not set
 # CONFIG_CRYPTO_MICHAEL_MIC is not set
 # CONFIG_CRYPTO_RMD128 is not set
 # CONFIG_CRYPTO_RMD160 is not set
 # CONFIG_CRYPTO_RMD256 is not set
 # CONFIG_CRYPTO_RMD320 is not set
-CONFIG_CRYPTO_SHA1=m
-CONFIG_CRYPTO_SHA256=m
+# CONFIG_CRYPTO_SHA1 is not set
+# CONFIG_CRYPTO_SHA256 is not set
 # CONFIG_CRYPTO_SHA512 is not set
 # CONFIG_CRYPTO_TGR192 is not set
 # CONFIG_CRYPTO_WP512 is not set
@@ -2072,14 +1296,14 @@ CONFIG_CRYPTO_SHA256=m
 #
 # Ciphers
 #
-CONFIG_CRYPTO_AES=y
+# CONFIG_CRYPTO_AES is not set
 # CONFIG_CRYPTO_ANUBIS is not set
 # CONFIG_CRYPTO_ARC4 is not set
 # CONFIG_CRYPTO_BLOWFISH is not set
 # CONFIG_CRYPTO_CAMELLIA is not set
 # CONFIG_CRYPTO_CAST5 is not set
 # CONFIG_CRYPTO_CAST6 is not set
-CONFIG_CRYPTO_DES=y
+# CONFIG_CRYPTO_DES is not set
 # CONFIG_CRYPTO_FCRYPT is not set
 # CONFIG_CRYPTO_KHAZAD is not set
 # CONFIG_CRYPTO_SALSA20 is not set
@@ -2099,8 +1323,7 @@ CONFIG_CRYPTO_LZO=y
 # Random Number Generation
 #
 # CONFIG_CRYPTO_ANSI_CPRNG is not set
-CONFIG_CRYPTO_HW=y
-CONFIG_CRYPTO_DEV_DCP=y
+# CONFIG_CRYPTO_HW is not set
 # CONFIG_BINARY_PRINTF is not set
 
 #
@@ -2108,13 +1331,13 @@ CONFIG_CRYPTO_DEV_DCP=y
 #
 CONFIG_BITREVERSE=y
 CONFIG_GENERIC_FIND_LAST_BIT=y
-CONFIG_CRC_CCITT=m
+# CONFIG_CRC_CCITT is not set
 CONFIG_CRC16=y
 # CONFIG_CRC_T10DIF is not set
 CONFIG_CRC_ITU_T=m
 CONFIG_CRC32=y
 # CONFIG_CRC7 is not set
-CONFIG_LIBCRC32C=m
+# CONFIG_LIBCRC32C is not set
 CONFIG_AUDIT_GENERIC=y
 CONFIG_ZLIB_INFLATE=y
 CONFIG_ZLIB_DEFLATE=y
-- 
1.7.4.1

From 550810b2999d0c31f8ea8a362ad681c794e1578b Mon Sep 17 00:00:00 2001
From: Nathaniel Houghton <nhoughto@qca.qualcomm.com>
Date: Tue, 21 Feb 2012 18:22:30 -0500
Subject: [PATCH] increase maximum RX frame size to 1522 bytes;
 drop frames that are too short or too long;

---
 linux-2.6.35.3/drivers/net/fec.c |   11 ++++++++++-
 1 files changed, 10 insertions(+), 1 deletions(-)

diff --git a/linux-2.6.35.3/drivers/net/fec.c b/drivers/net/fec.c
index 0838be9..27cf28e 100644
--- a/linux-2.6.35.3/drivers/net/fec.c
+++ b/linux-2.6.35.3/drivers/net/fec.c
@@ -127,11 +127,15 @@
 #define FEC_STOP_IMASK (FEC_ENET_MII)
 #endif
 
+#define FEC_DEFAULT_MAX_FL 1522
+#define FEC_RCR_FL_MAX_MASK  0x3FFF0000
+#define FEC_RCR_FL_MAX_SHIFT 16
+
 /* The FEC stores dest/src/type, data, and checksum for receive packets.
  */
 #define PKT_MAXBUF_SIZE		1518
 #define PKT_MINBUF_SIZE		64
-#define PKT_MAXBLR_SIZE		1520
+#define PKT_MAXBLR_SIZE		1536
 
 
 /*
@@ -549,6 +553,7 @@ fec_enet_rx(struct net_device *dev)
 			if (status & (BD_ENET_RX_LG | BD_ENET_RX_SH)) {
 				/* Frame too long or too short. */
 				dev->stats.rx_length_errors++;
+				goto rx_processing_done;
 			}
 			if (status & BD_ENET_RX_NO)	/* Frame alignment */
 				dev->stats.rx_frame_errors++;
@@ -1366,6 +1371,10 @@ fec_restart(struct net_device *dev, int duplex)
 	else
 		reg |= 0x00000200;  /* 10M */
 
+	/* Set maximum frame length to 1522 bytes. */
+	reg &= ~FEC_RCR_FL_MAX_MASK;
+	reg |= (FEC_DEFAULT_MAX_FL << FEC_RCR_FL_MAX_SHIFT) & FEC_RCR_FL_MAX_MASK;
+
 	writel(reg, fep->hwp + FEC_R_CNTRL);
 
 #endif
-- 
1.7.4.1

--- a/linux-2.6.35.3/arch/arm/configs/imx28evk_defconfig	2012-04-09 14:42:04.609317585 -0400
+++ b/linux-2.6.35.3/arch/arm/configs/imx28evk_defconfig	2012-04-09 14:54:07.036899942 -0400
@@ -466,7 +466,7 @@
 # CONFIG_L2TP is not set
 CONFIG_STP=y
 CONFIG_BRIDGE=y
-CONFIG_BRIDGE_IGMP_SNOOPING=y
+# CONFIG_BRIDGE_IGMP_SNOOPING is not set
 # CONFIG_NET_DSA is not set
 # CONFIG_VLAN_8021Q is not set
 # CONFIG_DECNET is not set
