diff --git a/qca-uart/qca_uart.c b/qca-uart/qca_uart.c
index 07230cf..07dedf0 100644
--- a/qca-uart/qca_uart.c
+++ b/qca-uart/qca_uart.c
@@ -46,6 +46,8 @@
 #include <linux/tty.h>
 #include <linux/version.h>
 
+#include <mach/qca_bootstrap.h>
+
 /*====================================================================*
  *   custom header files;
  *--------------------------------------------------------------------*/
@@ -492,6 +494,9 @@ qcauart_mod_init(void)
 	struct qcauart *qca = NULL;
 	int status;
 
+	/* reconfigure pins for UART */
+	qca_set_host_mode(QCA_HOST_MODE_UART);
+
 	printk(KERN_INFO "qcauart: version %s\n", QCAUART_VERSION);
 
 	/* Validate module parameters. */
